{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Part 0: Import\n",
    "**Import the various libraries**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from statistics import mean"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Part 1: Read Files\n",
    "**Raw data:**\n",
    "**(1) cycle 1 & cycle 35**\n",
    "**(2) genotype**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Please the AG sample number(ex: AG190604_A11_B12): AG191111_ANA18561_BNA18567_CNA18595_DNA18606\n",
      "Please enter the sample number(ex: A, B, C, D): c\n"
     ]
    }
   ],
   "source": [
    "################################\n",
    "# Read files: cycle 1 & cycle 35\n",
    "################################\n",
    "\n",
    "AG_num = str(input(\"Please the AG sample number(ex: AG190604_A11_B12): \"))\n",
    "sample_num = str(input(\"Please enter the sample number(ex: A, B, C, D): \"))\n",
    "#AG_num = 'AG190604_A11_B12'\n",
    "\n",
    "cycle1_dic = '../sample_data/' + AG_num +'-cycle 1.xls'\n",
    "cycle35_dic = '../sample_data/' + AG_num + '-cycle 35.xls'\n",
    "\n",
    "\n",
    "cycle1 = pd.read_excel(open(cycle1_dic, 'rb'),\n",
    "              sheet_name='Multicomponent Data',\n",
    "              header=None,\n",
    "              names = ['Well', 'Cycle', 'ROX', 'VIC', 'FAM'])\n",
    "\n",
    "cycle1 = pd.DataFrame(cycle1)\n",
    "cycle1 = cycle1.loc[47:]\n",
    "cycle1.head(50)\n",
    "\n",
    "cycle35 = pd.read_excel(open(cycle35_dic, 'rb'),\n",
    "              sheet_name='Multicomponent Data', \n",
    "              header=None,\n",
    "              names = ['Well', 'Cycle', 'ROX', 'VIC', 'FAM'])\n",
    "\n",
    "\n",
    "cycle35 = pd.DataFrame(cycle35)\n",
    "cycle35 = cycle35.loc[47:]\n",
    "cycle35.head(50)\n",
    "\n",
    "\n",
    "########################\n",
    "# Read file: AG genotype\n",
    "########################\n",
    "\n",
    "ag_genotype = pd.read_csv('../raw/AG_genotype.csv')\n",
    "\n",
    "# genotype 1\n",
    "gene1 = ag_genotype.iloc[:,7]\n",
    "\n",
    "# genotype 2\n",
    "gene2 = ag_genotype.iloc[:,8]\n",
    "\n",
    "# genotype 3\n",
    "gene3 = ag_genotype.iloc[:,9]\n",
    "\n",
    "#standard 1\n",
    "stand1 = ag_genotype.iloc[:,13]\n",
    "\n",
    "#standard 2\n",
    "stand2 = ag_genotype.iloc[:,14]\n",
    "\n",
    "# gene risk 1\n",
    "risk1 = ag_genotype.iloc[:,10]\n",
    "\n",
    "# gene risk 2\n",
    "risk2 = ag_genotype.iloc[:,11]\n",
    "\n",
    "# gene risk 3\n",
    "risk3 = ag_genotype.iloc[:,12]\n",
    "\n",
    "ag_genotype.head(87)\n",
    "\n",
    "\n",
    "###################\n",
    "# Read file: member\n",
    "###################\n",
    "\n",
    "member = pd.read_excel(open('../raw/來案暨出貨明細表-基因檢測.xlsx', 'rb'),\n",
    "              sheet_name='來案暨出貨明細表-基因檢測(201904起')\n",
    "\n",
    "#AG191111_ANA18561_BNA18567_CNA18595_DNA18606"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Membership"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get sample number for membership\n",
    "\n",
    "if sample_num.upper() == \"A\":\n",
    "    \n",
    "    index = AG_num[8:].find(\"A\") + 8\n",
    "    num = AG_num[(index + 1): (index + 3)]\n",
    "    \n",
    "    AG_sample = AG_num[0:8] + '-' + num\n",
    "\n",
    "    \n",
    "elif sample_num.upper() == \"B\":\n",
    "    \n",
    "    index = AG_num[8:].find(\"B\") + 8\n",
    "    num = AG_num[(index + 1): (index + 3)]\n",
    "    \n",
    "    AG_sample = AG_num[0:8] + '-' + num\n",
    "    \n",
    "    \n",
    "elif sample_num.upper() == \"C\":\n",
    "    \n",
    "    index = AG_num[8:].find(\"C\") + 8\n",
    "    num = AG_num[(index + 1): (index + 3)]\n",
    "    \n",
    "    AG_sample = AG_num[0:8] + '-' + num\n",
    "    \n",
    "    \n",
    "elif sample_num.upper() == \"D\":\n",
    "    \n",
    "    index = AG_num[8:].find(\"D\") + 8\n",
    "    num = AG_num[(index + 1): (index + 3)]\n",
    "    \n",
    "    AG_sample = AG_num[0:8] + '-' + num"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>訂單編號</th>\n",
       "      <th>檢測部出貨\n",
       "請填日期</th>\n",
       "      <th>業務出貨                          請填日期</th>\n",
       "      <th>來案人</th>\n",
       "      <th>客服</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>醫療院所          (報告書抬頭)</th>\n",
       "      <th>家屬關係</th>\n",
       "      <th>來案日期</th>\n",
       "      <th>預定寄出日</th>\n",
       "      <th>...</th>\n",
       "      <th>Unnamed: 13</th>\n",
       "      <th>以前做過檢驗項目</th>\n",
       "      <th>檢體編號</th>\n",
       "      <th>報告型式</th>\n",
       "      <th>報告語言</th>\n",
       "      <th>BS報告書DM需求   請打V</th>\n",
       "      <th>異常</th>\n",
       "      <th>備註</th>\n",
       "      <th>身高</th>\n",
       "      <th>體重</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [訂單編號, 檢測部出貨\n",
       "請填日期, 業務出貨                          請填日期  , 來案人, 客服, Unnamed: 5, 醫療院所          (報告書抬頭), 家屬關係, 來案日期, 預定寄出日, 會員編號\n",
       "(系統匯出), 顧客姓名, 檢測專案\n",
       "(訂購表查詢), Unnamed: 13, 以前做過檢驗項目, 檢體編號, 報告型式, 報告語言, BS報告書DM需求   請打V, 異常, 備註, 身高, 體重]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 23 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "member[(member[['檢體編號']] == AG_sample).all(1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "ename": "IndexError",
     "evalue": "single positional indexer is out-of-bounds",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-5-fd1135c748f1>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# member number\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mmember_number\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmember\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmember\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'檢體編號'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mAG_sample\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m10\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;31m# report date\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mdate\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmember\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmember\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'檢體編號'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mAG_sample\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m9\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m'/'\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmember\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmember\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'檢體編號'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mAG_sample\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m9\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;36m7\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m'/'\u001b[0m \u001b[0;34m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmember\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmember\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'檢體編號'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mAG_sample\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0miloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m9\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m8\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;36m10\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   1492\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mKeyError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mIndexError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mAttributeError\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1493\u001b[0m                 \u001b[0;32mpass\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1494\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_tuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1495\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1496\u001b[0m             \u001b[0;31m# we by definition only have the 0th axis\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_getitem_tuple\u001b[0;34m(self, tup)\u001b[0m\n\u001b[1;32m   2141\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_getitem_tuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2142\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2143\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_has_valid_tuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2144\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2145\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_lowerdim\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_has_valid_tuple\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m    221\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0mIndexingError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Too many indexers'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    222\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 223\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_validate_key\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    224\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    225\u001b[0m                 raise ValueError(\"Location based indexing can only have \"\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_validate_key\u001b[0;34m(self, key, axis)\u001b[0m\n\u001b[1;32m   2068\u001b[0m             \u001b[0;32mreturn\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2069\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0mis_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2070\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_validate_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2071\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtuple\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2072\u001b[0m             \u001b[0;31m# a tuple should already have been caught by this point\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_validate_integer\u001b[0;34m(self, key, axis)\u001b[0m\n\u001b[1;32m   2137\u001b[0m         \u001b[0mlen_axis\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2138\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mkey\u001b[0m \u001b[0;34m>=\u001b[0m \u001b[0mlen_axis\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mkey\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;34m-\u001b[0m\u001b[0mlen_axis\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2139\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mIndexError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"single positional indexer is out-of-bounds\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2140\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2141\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_getitem_tuple\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mIndexError\u001b[0m: single positional indexer is out-of-bounds"
     ]
    }
   ],
   "source": [
    "# member number\n",
    "member_number = member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,10]\n",
    "\n",
    "# report date\n",
    "date = str(member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,9])[0:4] + '/' + str(member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,9])[5:7] + '/' +str(member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,9])[8:10]\n",
    "\n",
    "# gender\n",
    "gender = member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,13]\n",
    "\n",
    "# paper or electronic\n",
    "report_type = member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,16]\n",
    "\n",
    "# language\n",
    "language = member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,17]\n",
    "\n",
    "\n",
    "# clinic\n",
    "\n",
    "if member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,6] == 0:\n",
    "    clinic = \"健樂士診所\"\n",
    "\n",
    "#elif math.isnan(member[(member[['檢體編號']] == G2_num).all(1)].iloc[0,6]):\n",
    "#    clinic = \"健樂士診所\"\n",
    "\n",
    "elif member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,6] == '':\n",
    "    clinic = \"健樂士診所\"\n",
    "\n",
    "elif member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,6] != '':\n",
    "    clinic = member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,6]\n",
    "\n",
    "\n",
    "\n",
    "# clinic SC\n",
    "\n",
    "if member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,6] == 0:\n",
    "    clinic_SC = \"健乐士诊所\"\n",
    "\n",
    "#elif math.isnan(member[(member[['檢體編號']] == G2_num).all(1)].iloc[0,6]):\n",
    "#    clinic_SC = \"健乐士诊所\"\n",
    "\n",
    "elif member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,6] == '':\n",
    "    clinic_SC = \"健乐士诊所\"\n",
    "\n",
    "elif member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,6] != '':\n",
    "    clinic_SC = member[(member[['檢體編號']] == AG_sample).all(1)].iloc[0,6]\n",
    "\n",
    "\n",
    "membership = {'data':['sample ID', 'clinic', 'member ID', 'date', 'gender', 'report_type', 'language'], 'info':[AG_sample, clinic, member_number, date, gender, report_type, language]}\n",
    "\n",
    "membership_SC = {'data':['sample ID', 'clinic', 'member ID', 'date', 'gender', 'report_type', 'language'], 'info':[AG_sample, clinic_SC, member_number, date, gender, report_type, language]}\n",
    "\n",
    "# Create DataFrame\n",
    "membership = pd.DataFrame(membership)\n",
    "membership\n",
    "\n",
    "# write csv\n",
    "outname = AG_sample + '_membership.csv'\n",
    "\n",
    "outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)\n",
    "\n",
    "fullname = os.path.join(outdir, outname)\n",
    "\n",
    "membership.to_csv(fullname , encoding='utf-8')\n",
    "\n",
    "print(AG_sample + ' finish output ' + fullname + '\\n')\n",
    "\n",
    "\n",
    "# membership_SC\n",
    "\n",
    "membership_SC = pd.DataFrame(membership_SC)\n",
    "membership_SC\n",
    "\n",
    "# write csv\n",
    "outname = AG_sample + '_membership_SC.csv'\n",
    "\n",
    "outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)\n",
    "\n",
    "fullname = os.path.join(outdir, outname)\n",
    "\n",
    "membership_SC.to_csv(fullname , encoding='utf-8')\n",
    "\n",
    "\n",
    "print(AG_sample + ' finish output ' + fullname + '\\n')\n",
    "\n",
    "print(\"\")\n",
    "print(\"---------------------------\")\n",
    "print('sample ID: ', AG_sample)\n",
    "print('clinic: ', clinic)\n",
    "print('member ID: ', member_number)\n",
    "print('report date: ', date)\n",
    "print('gender: ', gender)\n",
    "print('report_type: ', report_type)\n",
    "print('language: ', language)\n",
    "print(\"---------------------------\")\n",
    "print(\"\")\n",
    "\n",
    "membership"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Part 2: Calculate the vic & fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2) AG191111_ANA18561_BNA18567_CNA18595_DNA18606 get vic and fam value.\n",
      "\n",
      "(3) AG191111_ANA18561_BNA18567_CNA18595_DNA18606 finish delta_vic calculation.\n",
      "\n",
      "(4) AG191111_ANA18561_BNA18567_CNA18595_DNA18606 finish delta_fam calculation.\n",
      "\n",
      "(5) AG191111_ANA18561_BNA18567_CNA18595_DNA18606 finish delta vic and fam ratio calculate.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#################################\n",
    "# calculate delta_vic & delta_fam\n",
    "\n",
    "delta_vic = list(cycle35['VIC'] - cycle1['VIC'])\n",
    "#print(delta_vic)\n",
    "\n",
    "delta_fam = list(cycle35['FAM'] - cycle1['FAM'])\n",
    "#print(delta_fam)\n",
    "\n",
    "delta = pd.DataFrame(list(zip(delta_vic, delta_fam)), \n",
    "                    columns=['delta_vic','delta_fam'])\n",
    "\n",
    "    \n",
    "    \n",
    "delta.head()\n",
    "\n",
    "print('(2) ' + AG_num + ' get vic and fam value.\\n')\n",
    "\n",
    "\n",
    "\n",
    "#############################\n",
    "# get the signal of delta_vic\n",
    "\n",
    "vic_signal = []\n",
    "\n",
    "for cell in delta_vic:\n",
    "    if cell >= 500:\n",
    "        vic_signal.append(1)\n",
    "        \n",
    "    else:\n",
    "        vic_signal.append(0)\n",
    "    \n",
    "#print(vic_signal)\n",
    "\n",
    "\n",
    "print('(3) ' + AG_num + ' finish delta_vic calculation.\\n')\n",
    "\n",
    "\n",
    "#############################\n",
    "# get the signal of delta_fam\n",
    "\n",
    "fam_signal = []\n",
    "\n",
    "for cell in delta_fam:\n",
    "    if cell >= 500:\n",
    "        fam_signal.append(1)\n",
    "        \n",
    "    else:\n",
    "        fam_signal.append(0)\n",
    "    \n",
    "#print(fam_signal)\n",
    "\n",
    "\n",
    "print('(4) ' + AG_num + ' finish delta_fam calculation.\\n')\n",
    "\n",
    "\n",
    "########################################\n",
    "# get the error of delta_vic & delta_fam\n",
    "\n",
    "signal = []\n",
    "\n",
    "for i in range(0,len(vic_signal)):\n",
    "    signal.append(vic_signal[i] + fam_signal[i])\n",
    "\n",
    "#print(signal)\n",
    "\n",
    "\n",
    "error = []\n",
    "    \n",
    "for cell in signal:\n",
    "    if cell == 2:\n",
    "        error.append(0)\n",
    "    elif cell != 2:\n",
    "        error.append(1)\n",
    "\n",
    "\n",
    "        \n",
    "######################################\n",
    "# get the value: delta_vic / delta_fam\n",
    "\n",
    "value = []\n",
    "\n",
    "for i in range(len(delta_vic )):\n",
    "    \n",
    "    if vic_signal[i] == 1 and fam_signal[i] == 1:\n",
    "        \n",
    "        value.append(delta_vic[i] / delta_fam[i])\n",
    "    \n",
    "    elif vic_signal[i] != 1 and fam_signal[i] == 1:\n",
    "        \n",
    "        value.append(1 / delta_fam[i])\n",
    "\n",
    "    elif vic_signal[i] == 1 and fam_signal[i] != 1:\n",
    "        \n",
    "        value.append(delta_vic[i] / 1)\n",
    "        \n",
    "    else:\n",
    "        value.append(1)\n",
    "\n",
    "#print(value)\n",
    "\n",
    "print('(5) ' + AG_num + ' finish delta vic and fam ratio calculate.\\n')\n",
    "\n",
    "\n",
    "\n",
    "value_vic_fam = pd.DataFrame(list(zip(delta_vic, delta_fam, vic_signal, fam_signal, signal, error, value)), \n",
    "                    columns=['delta_vic','delta_fam', 'vic_signal', 'fam_signal', 'sum_signal', 'error', 'value'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Part 3: Calculate the genotype: A-D"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### A-D genotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AG191111_ANA18561_BNA18567_CNA18595_DNA18606 finish calculate the \"c_value\".\n",
      "\n",
      "AG191111_ANA18561_BNA18567_CNA18595_DNA18606 finish calculate \"c_genotype\".\n",
      "\n",
      "AG191111_ANA18561_BNA18567_CNA18595_DNA18606 finish calculate \"c_score\". \n",
      "\n",
      "AG191111_ANA18561_BNA18567_CNA18595_DNA18606 finish calculate \"c_result\". \n",
      "\n",
      "AG191111_ANA18561_BNA18567_CNA18595_DNA18606 finish output ../result/191111/AG191111-NA/AG191111_ANA18561_BNA18567_CNA18595_DNA18606_c_result.csv\n",
      "\n",
      "AG191111_ANA18561_BNA18567_CNA18595_DNA18606 finish output ../result/191111/AG191111-NA/AG191111_ANA18561_BNA18567_CNA18595_DNA18606_c_category.csv\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if sample_num.upper() == \"A\":\n",
    "    \n",
    "    # get value (delta vic / delta fam)\n",
    "\n",
    "    a_value = value_vic_fam.iloc[[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,\n",
    "                        24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35,\n",
    "                        48, 49, \n",
    "                        35,\n",
    "                        50, 51, 52, 53, 54, 55, 56,\n",
    "                        51, 50,\n",
    "                        57, 58, 59,\n",
    "                        72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83,\n",
    "                        96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107,\n",
    "                        120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131,\n",
    "                        144, 145, 146, 147, 148, 149, 150, 151, 153, 152, 154, 155,\n",
    "                        168, 169, 170, 171, 172, 173, 174, 175, 176, 177], [6]]\n",
    "\n",
    "    a_ratio = []\n",
    "\n",
    "    for row in range(len(a_value)):\n",
    "\n",
    "        a_ratio.append(a_value.iloc[row,0])\n",
    "\n",
    "    print(AG_num + ' finish calculate the \"a_value\".' + '\\n')\n",
    "\n",
    "    #a_ratio\n",
    "\n",
    "\n",
    "    # Get genotype\n",
    "\n",
    "    a_genotype = []\n",
    "\n",
    "    for row in range(len(a_value)):\n",
    "\n",
    "        if stand1[row] > stand2[row]:\n",
    "\n",
    "            if float(a_value.iloc[row,]) > stand1[row]:\n",
    "\n",
    "                a_genotype.append(gene1[row])\n",
    "\n",
    "            elif float(a_value.iloc[row,]) < stand2[row]:\n",
    "\n",
    "                a_genotype.append(gene3[row])\n",
    "\n",
    "            else:\n",
    "\n",
    "                a_genotype.append(gene2[row])  \n",
    "\n",
    "        elif stand1[row] < stand2[row]:\n",
    "\n",
    "            if float(a_value.iloc[row,]) > stand2[row]:\n",
    "\n",
    "                a_genotype.append(gene3[row])\n",
    "\n",
    "            elif float(a_value.iloc[row,]) < stand1[row]:\n",
    "\n",
    "                a_genotype.append(gene1[row])\n",
    "\n",
    "            else:\n",
    "\n",
    "                a_genotype.append(gene2[row])  \n",
    "\n",
    "\n",
    "    print(AG_num + ' finish calculate \"a_genotype\".' +'\\n')\n",
    "\n",
    "    #a_genotype\n",
    "\n",
    "\n",
    "\n",
    "    # A score\n",
    "\n",
    "    a_score = []\n",
    "\n",
    "    for row in range(len(a_genotype)):\n",
    "\n",
    "        if a_genotype[row] == gene1[row]:\n",
    "\n",
    "            a_score.append(40*risk1[row] + 20)\n",
    "\n",
    "        elif a_genotype[row] == gene2[row]:\n",
    "\n",
    "            a_score.append(20*risk2[row] + 60)\n",
    "\n",
    "        elif a_genotype[row] == gene3[row]:\n",
    "\n",
    "            a_score.append(20*risk3[row] + 80)\n",
    "\n",
    "    print(AG_num + ' finish calculate \"a_score\". ' +'\\n')   \n",
    "\n",
    "\n",
    "    # A result\n",
    "\n",
    "    #############\n",
    "    # (1) 細胞衰老\n",
    "\n",
    "    # 1.細胞循環\n",
    "\n",
    "    g1_1 = mean(a_score[0:5])\n",
    "\n",
    "    # 2.DNA損傷檢查點\n",
    "\n",
    "    g1_2 = mean(a_score[5:10])\n",
    "\n",
    "    # 3.端粒酶活性\n",
    "\n",
    "    g1_3 = mean(a_score[10:12])\n",
    "\n",
    "    # 4.DNA修復\n",
    "\n",
    "    g1_4 = mean(a_score[12:20])\n",
    "\n",
    "    # 5.粒線體活性\n",
    "\n",
    "    g1_5 = mean(a_score[20:26])\n",
    "\n",
    "    # 6.對抗自由基\n",
    "\n",
    "    g1_6 = mean(a_score[26:32])\n",
    "\n",
    "\n",
    "    #############\n",
    "    # (2) 毒物代謝\n",
    "\n",
    "    # 1.肝臟\n",
    "\n",
    "    g2_1 = mean(a_score[32:37])\n",
    "\n",
    "    # 2.肺臟\n",
    "\n",
    "    g2_2 = mean(a_score[37:40])\n",
    "\n",
    "    # 3.腸道\n",
    "\n",
    "    g2_3 = mean(a_score[40:43])\n",
    "\n",
    "    # 4.腎臟\n",
    "\n",
    "    g2_4 = mean(a_score[43:47])\n",
    "\n",
    "\n",
    "    #############\n",
    "    # (3) 營養代謝\n",
    "\n",
    "    # 1.血管健康\n",
    "\n",
    "    g3_1 = mean(a_score[47:51])\n",
    "\n",
    "    # 2.蛋白質代謝\n",
    "\n",
    "    g3_2 = mean(a_score[51:57])\n",
    "\n",
    "    # 3.醣類代謝\n",
    "\n",
    "    g3_3 = mean(a_score[57:65])\n",
    "\n",
    "    # 4.脂質代謝\n",
    "\n",
    "    g3_4 = mean(a_score[65:72])\n",
    "\n",
    "\n",
    "    #############\n",
    "    # (4) 免疫調節\n",
    "\n",
    "    # 1.抗原呈現\n",
    "\n",
    "    g4_1 = mean(a_score[72:77])\n",
    "\n",
    "    # 2.免疫訓練\n",
    "\n",
    "    g4_2 = mean(a_score[77:88])\n",
    "\n",
    "    # 3.發炎因子\n",
    "\n",
    "    g4_3 = mean(a_score[88:97])\n",
    "\n",
    "\n",
    "\n",
    "    #####################\n",
    "    # Score of 4 category\n",
    "\n",
    "    # (1) 細胞衰老\n",
    "\n",
    "    g1 = mean(a_score[0:32])\n",
    "\n",
    "    # (2) 毒物代謝\n",
    "\n",
    "    g2 = mean(a_score[32:47])\n",
    "\n",
    "    # (3) 營養代謝\n",
    "\n",
    "    g3 = mean(a_score[47:72])\n",
    "\n",
    "    # (4) 免疫調節\n",
    "\n",
    "    g4 = mean(a_score[72:97])\n",
    "\n",
    "\n",
    "    #############################################\n",
    "    # Pick the bad subcategory from each category\n",
    "\n",
    "\n",
    "    sub = []\n",
    "\n",
    "    # (1) 細胞衰老\n",
    "\n",
    "    if min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_1:\n",
    "\n",
    "        sub.append(\"細胞循環\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_2:\n",
    "\n",
    "        sub.append(\"DNA損傷檢查點\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_3:\n",
    "\n",
    "        sub.append(\"端粒酶活性\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_4:\n",
    "\n",
    "        sub.append(\"DNA修復\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_5:\n",
    "\n",
    "        sub.append(\"粒線體活性\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_6:\n",
    "\n",
    "        sub.append(\"對抗自由基\")\n",
    "\n",
    "\n",
    "    # (2) 毒物代謝\n",
    "\n",
    "    if min(g2_1, g2_2, g2_3, g2_4) == g2_1:\n",
    "\n",
    "        sub.append(\"肝臟\")\n",
    "\n",
    "    elif min(g2_1, g2_2, g2_3, g2_4) == g2_2:\n",
    "\n",
    "        sub.append(\"肺臟\")\n",
    "\n",
    "    elif min(g2_1, g2_2, g2_3, g2_4) == g2_3:\n",
    "\n",
    "        sub.append(\"腸道\")\n",
    "\n",
    "    elif min(g2_1, g2_2, g2_3, g2_4) == g2_4:\n",
    "\n",
    "        sub.append(\"腎臟\")\n",
    "\n",
    "\n",
    "    # (3) 營養代謝\n",
    "\n",
    "    if min(g3_1, g3_2, g3_3, g3_4) == g3_1:\n",
    "\n",
    "        sub.append(\"血管健康\")\n",
    "\n",
    "    elif min(g3_1, g3_2, g3_3, g3_4) == g3_2:\n",
    "\n",
    "        sub.append(\"蛋白質代謝\")\n",
    "\n",
    "    elif min(g3_1, g3_2, g3_3, g3_4) == g3_3:\n",
    "\n",
    "        sub.append(\"醣類代謝\")\n",
    "\n",
    "    elif min(g3_1, g3_2, g3_3, g3_4) == g3_4:\n",
    "\n",
    "        sub.append(\"脂質代謝\")\n",
    "\n",
    "\n",
    "    # (4) 免疫調節\n",
    "\n",
    "    if min(g4_1, g4_2, g4_3) == g4_1:\n",
    "\n",
    "        sub.append(\"抗原呈現\")\n",
    "\n",
    "    elif min(g4_1, g4_2, g4_3) == g4_2:\n",
    "\n",
    "        sub.append(\"免疫訓練\")\n",
    "\n",
    "    elif min(g4_1, g4_2, g4_3) == g4_3:\n",
    "\n",
    "        sub.append(\"發炎因子\")\n",
    "\n",
    "\n",
    "    print(AG_num + ' finish calculate \"a_result\". ' +'\\n')\n",
    "\n",
    "    sub\n",
    "\n",
    "    #####################\n",
    "    # output a_result.csv\n",
    "    \n",
    "    ag_genotype['a_value'] = a_ratio\n",
    "    ag_genotype['a_genotype'] = a_genotype\n",
    "    ag_genotype['a_score'] = a_score\n",
    "\n",
    "    ag_genotype\n",
    "\n",
    "\n",
    "    outname = AG_num + '_a_result.csv'\n",
    "\n",
    "    outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "    if not os.path.exists(outdir):\n",
    "        os.makedirs(outdir)\n",
    "\n",
    "    fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "    ag_genotype.to_csv(fullname)\n",
    "\n",
    "    print(AG_num + ' finish output ' + fullname + '\\n')\n",
    "    \n",
    "    \n",
    "    #######################\n",
    "    # output a_category.csv\n",
    "    \n",
    "    category = [\"細胞衰老\", \"毒物代謝\", \"營養代謝\", \"免疫調節\"]\n",
    "\n",
    "    a_mean = [format(g1, '.0f'), format(g2, '.0f'), format(g3, '.0f'), format(g4, '.0f')]\n",
    "\n",
    "    a_category = pd.DataFrame(list(zip(category, a_mean, sub)),\n",
    "                  columns=['category','mean', 'high risk subcategory'])\n",
    "\n",
    "\n",
    "    outname = AG_num + '_a_category.csv'\n",
    "\n",
    "    outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "    if not os.path.exists(outdir):\n",
    "        os.makedirs(outdir)\n",
    "\n",
    "    fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "    a_category.to_csv(fullname)\n",
    "\n",
    "    print(AG_num + ' finish output ' + fullname + '\\n')\n",
    "    \n",
    "    \n",
    "    \n",
    "elif sample_num.upper() == \"B\":\n",
    "    \n",
    "    # get value (delta vic / delta fam)\n",
    "\n",
    "    b_value = value_vic_fam.iloc[[12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23,\n",
    "                        36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47,\n",
    "                        60, 61, \n",
    "                        47,\n",
    "                        62, 63, 64, 65, 66, 67, 68,\n",
    "                        63, 62,\n",
    "                        69, 70, 71,\n",
    "                        84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95,\n",
    "                        108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119,\n",
    "                        132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143,\n",
    "                        156, 157, 158, 159, 160, 161, 162, 163, 165, 164, 166, 167,\n",
    "                        180, 181, 182, 183, 184, 185, 186, 187, 188, 189], [6]]\n",
    "\n",
    "    b_ratio = []\n",
    "\n",
    "    for row in range(len(b_value)):\n",
    "\n",
    "        b_ratio.append(b_value.iloc[row,0])\n",
    "\n",
    "    print(AG_num + ' finish calculate the \"b_value\".' + '\\n')\n",
    "\n",
    "\n",
    "\n",
    "    # Get genotype\n",
    "\n",
    "    b_genotype = []\n",
    "\n",
    "    for row in range(len(b_value)):\n",
    "\n",
    "        if stand1[row] > stand2[row]:\n",
    "\n",
    "            if float(b_value.iloc[row,]) > stand1[row]:\n",
    "\n",
    "                b_genotype.append(gene1[row])\n",
    "\n",
    "            elif float(b_value.iloc[row,]) < stand2[row]:\n",
    "\n",
    "                b_genotype.append(gene3[row])\n",
    "\n",
    "            else:\n",
    "\n",
    "                b_genotype.append(gene2[row])  \n",
    "\n",
    "        elif stand1[row] < stand2[row]:\n",
    "\n",
    "            if float(b_value.iloc[row,]) > stand2[row]:\n",
    "\n",
    "                b_genotype.append(gene3[row])\n",
    "\n",
    "            elif float(b_value.iloc[row,]) < stand1[row]:\n",
    "\n",
    "                b_genotype.append(gene1[row])\n",
    "\n",
    "            else:\n",
    "\n",
    "                b_genotype.append(gene2[row])  \n",
    "\n",
    "\n",
    "    print(AG_num + ' finish calculate \"b_genotype\".' +'\\n')\n",
    "\n",
    "    #b_genotype\n",
    "\n",
    "\n",
    "\n",
    "    # B score\n",
    "\n",
    "    b_score = []\n",
    "\n",
    "    for row in range(len(b_genotype)):\n",
    "\n",
    "        if b_genotype[row] == gene1[row]:\n",
    "\n",
    "            b_score.append(40*risk1[row] + 20)\n",
    "\n",
    "        elif b_genotype[row] == gene2[row]:\n",
    "\n",
    "            b_score.append(20*risk2[row] + 60)\n",
    "\n",
    "        elif b_genotype[row] == gene3[row]:\n",
    "\n",
    "            b_score.append(20*risk3[row] + 80)\n",
    "\n",
    "    print(AG_num + ' finish calculate \"b_score\". ' +'\\n')   \n",
    "\n",
    "\n",
    "\n",
    "    # B result\n",
    "\n",
    "    #############\n",
    "    # (1) 細胞衰老\n",
    "\n",
    "    # 1.細胞循環\n",
    "\n",
    "    g1_1 = mean(b_score[0:5])\n",
    "\n",
    "    # 2.DNA損傷檢查點\n",
    "\n",
    "    g1_2 = mean(b_score[5:10])\n",
    "\n",
    "    # 3.端粒酶活性\n",
    "\n",
    "    g1_3 = mean(b_score[10:12])\n",
    "\n",
    "    # 4.DNA修復\n",
    "\n",
    "    g1_4 = mean(b_score[12:20])\n",
    "\n",
    "    # 5.粒線體活性\n",
    "\n",
    "    g1_5 = mean(b_score[20:26])\n",
    "\n",
    "    # 6.對抗自由基\n",
    "\n",
    "    g1_6 = mean(b_score[26:32])\n",
    "\n",
    "\n",
    "    #############\n",
    "    # (2) 毒物代謝\n",
    "\n",
    "    # 1.肝臟\n",
    "\n",
    "    g2_1 = mean(b_score[32:37])\n",
    "\n",
    "    # 2.肺臟\n",
    "\n",
    "    g2_2 = mean(b_score[37:40])\n",
    "\n",
    "    # 3.腸道\n",
    "\n",
    "    g2_3 = mean(b_score[40:43])\n",
    "\n",
    "    # 4.腎臟\n",
    "\n",
    "    g2_4 = mean(b_score[43:47])\n",
    "\n",
    "\n",
    "    #############\n",
    "    # (3) 營養代謝\n",
    "\n",
    "    # 1.血管健康\n",
    "\n",
    "    g3_1 = mean(b_score[47:51])\n",
    "\n",
    "    # 2.蛋白質代謝\n",
    "\n",
    "    g3_2 = mean(b_score[51:57])\n",
    "\n",
    "    # 3.醣類代謝\n",
    "\n",
    "    g3_3 = mean(b_score[57:65])\n",
    "\n",
    "    # 4.脂質代謝\n",
    "\n",
    "    g3_4 = mean(b_score[65:72])\n",
    "\n",
    "\n",
    "    #############\n",
    "    # (4) 免疫調節\n",
    "\n",
    "    # 1.抗原呈現\n",
    "\n",
    "    g4_1 = mean(b_score[72:77])\n",
    "\n",
    "    # 2.免疫訓練\n",
    "\n",
    "    g4_2 = mean(b_score[77:88])\n",
    "\n",
    "    # 3.發炎因子\n",
    "\n",
    "    g4_3 = mean(b_score[88:97])\n",
    "\n",
    "\n",
    "\n",
    "    #####################\n",
    "    # Score of 4 category\n",
    "\n",
    "    # (1) 細胞衰老\n",
    "\n",
    "    g1 = mean(b_score[0:32])\n",
    "\n",
    "    # (2) 毒物代謝\n",
    "\n",
    "    g2 = mean(b_score[32:47])\n",
    "\n",
    "    # (3) 營養代謝\n",
    "\n",
    "    g3 = mean(b_score[47:72])\n",
    "\n",
    "    # (4) 免疫調節\n",
    "\n",
    "    g4 = mean(b_score[72:97])\n",
    "\n",
    "\n",
    "    #############################################\n",
    "    # Pick the bad subcategory from each category\n",
    "\n",
    "\n",
    "    sub = []\n",
    "\n",
    "    # (1) 細胞衰老\n",
    "\n",
    "    if min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_1:\n",
    "\n",
    "        sub.append(\"細胞循環\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_2:\n",
    "\n",
    "        sub.append(\"DNA損傷檢查點\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_3:\n",
    "\n",
    "        sub.append(\"端粒酶活性\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_4:\n",
    "\n",
    "        sub.append(\"DNA修復\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_5:\n",
    "\n",
    "        sub.append(\"粒線體活性\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_6:\n",
    "\n",
    "        sub.append(\"對抗自由基\")\n",
    "\n",
    "\n",
    "    # (2) 毒物代謝\n",
    "\n",
    "    if min(g2_1, g2_2, g2_3, g2_4) == g2_1:\n",
    "\n",
    "        sub.append(\"肝臟\")\n",
    "\n",
    "    elif min(g2_1, g2_2, g2_3, g2_4) == g2_2:\n",
    "\n",
    "        sub.append(\"肺臟\")\n",
    "\n",
    "    elif min(g2_1, g2_2, g2_3, g2_4) == g2_3:\n",
    "\n",
    "        sub.append(\"腸道\")\n",
    "\n",
    "    elif min(g2_1, g2_2, g2_3, g2_4) == g2_4:\n",
    "\n",
    "        sub.append(\"腎臟\")\n",
    "\n",
    "\n",
    "    # (3) 營養代謝\n",
    "\n",
    "    if min(g3_1, g3_2, g3_3, g3_4) == g3_1:\n",
    "\n",
    "        sub.append(\"血管健康\")\n",
    "\n",
    "    elif min(g3_1, g3_2, g3_3, g3_4) == g3_2:\n",
    "\n",
    "        sub.append(\"蛋白質代謝\")\n",
    "\n",
    "    elif min(g3_1, g3_2, g3_3, g3_4) == g3_3:\n",
    "\n",
    "        sub.append(\"醣類代謝\")\n",
    "\n",
    "    elif min(g3_1, g3_2, g3_3, g3_4) == g3_4:\n",
    "\n",
    "        sub.append(\"脂質代謝\")\n",
    "\n",
    "\n",
    "    # (4) 免疫調節\n",
    "\n",
    "    if min(g4_1, g4_2, g4_3) == g4_1:\n",
    "\n",
    "        sub.append(\"抗原呈現\")\n",
    "\n",
    "    elif min(g4_1, g4_2, g4_3) == g4_2:\n",
    "\n",
    "        sub.append(\"免疫訓練\")\n",
    "\n",
    "    elif min(g4_1, g4_2, g4_3) == g4_3:\n",
    "\n",
    "        sub.append(\"發炎因子\")\n",
    "\n",
    "\n",
    "    print(AG_num + ' finish calculate \"b_result\". ' +'\\n')\n",
    "\n",
    "    sub\n",
    "    \n",
    "    #####################\n",
    "    # output b_result.csv\n",
    "\n",
    "    ag_genotype['b_value'] = b_ratio\n",
    "    ag_genotype['b_genotype'] = b_genotype\n",
    "    ag_genotype['b_score'] = b_score\n",
    "\n",
    "    ag_genotype\n",
    "\n",
    "\n",
    "    outname = AG_num + '_b_result.csv'\n",
    "\n",
    "    outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "    if not os.path.exists(outdir):\n",
    "        os.makedirs(outdir)\n",
    "\n",
    "    fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "    ag_genotype.to_csv(fullname)\n",
    "\n",
    "    print(AG_num + ' finish output ' + fullname + '\\n')\n",
    "    \n",
    "    \n",
    "    #######################\n",
    "    # output b_category.csv\n",
    "    \n",
    "    category = [\"細胞衰老\", \"毒物代謝\", \"營養代謝\", \"免疫調節\"]\n",
    "\n",
    "    b_mean = [format(g1, '.0f'), format(g2, '.0f'), format(g3, '.0f'), format(g4, '.0f')]\n",
    "\n",
    "    b_category = pd.DataFrame(list(zip(category, b_mean, sub)),\n",
    "                  columns=['category','mean', 'high risk subcategory'])\n",
    "\n",
    "\n",
    "    outname = AG_num + '_b_category.csv'\n",
    "\n",
    "    outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "    if not os.path.exists(outdir):\n",
    "        os.makedirs(outdir)\n",
    "\n",
    "    fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "    b_category.to_csv(fullname)\n",
    "\n",
    "    print(AG_num + ' finish output ' + fullname + '\\n')\n",
    "    \n",
    "    \n",
    "    \n",
    "elif sample_num.upper() == \"C\":\n",
    "    \n",
    "    # get value (delta vic / delta fam)\n",
    "\n",
    "    c_value = value_vic_fam.iloc[[192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203,\n",
    "                        216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227,\n",
    "                        240, 241, \n",
    "                        227,\n",
    "                        242, 243, 244, 245, 246, 247, 248,\n",
    "                        243, 242,\n",
    "                        249, 250, 251,\n",
    "                        264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275,\n",
    "                        288, 289, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299,\n",
    "                        312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323,\n",
    "                        336, 337, 338, 339, 340, 341, 342, 343, 345, 344, 346, 347,\n",
    "                        360, 361, 362, 363, 364, 365, 366, 367, 368, 369], [6]]\n",
    "\n",
    "    c_ratio = []\n",
    "\n",
    "    for row in range(len(c_value)):\n",
    "\n",
    "        c_ratio.append(c_value.iloc[row,0])\n",
    "\n",
    "    print(AG_num + ' finish calculate the \"c_value\".' + '\\n')\n",
    "\n",
    "\n",
    "    # Get genotype\n",
    "\n",
    "    c_genotype = []\n",
    "\n",
    "    for row in range(len(c_value)):\n",
    "\n",
    "        if stand1[row] > stand2[row]:\n",
    "\n",
    "            if float(c_value.iloc[row,]) > stand1[row]:\n",
    "\n",
    "                c_genotype.append(gene1[row])\n",
    "\n",
    "            elif float(c_value.iloc[row,]) < stand2[row]:\n",
    "\n",
    "                c_genotype.append(gene3[row])\n",
    "\n",
    "            else:\n",
    "\n",
    "                c_genotype.append(gene2[row])  \n",
    "\n",
    "        elif stand1[row] < stand2[row]:\n",
    "\n",
    "            if float(c_value.iloc[row,]) > stand2[row]:\n",
    "\n",
    "                c_genotype.append(gene3[row])\n",
    "\n",
    "            elif float(c_value.iloc[row,]) < stand1[row]:\n",
    "\n",
    "                c_genotype.append(gene1[row])\n",
    "\n",
    "            else:\n",
    "\n",
    "                c_genotype.append(gene2[row])  \n",
    "\n",
    "\n",
    "    print(AG_num + ' finish calculate \"c_genotype\".' +'\\n')\n",
    "\n",
    "    #c_genotype\n",
    "\n",
    "\n",
    "\n",
    "    # C score\n",
    "\n",
    "    c_score = []\n",
    "\n",
    "    for row in range(len(c_genotype)):\n",
    "\n",
    "        if c_genotype[row] == gene1[row]:\n",
    "\n",
    "            c_score.append(40*risk1[row] + 20)\n",
    "\n",
    "        elif c_genotype[row] == gene2[row]:\n",
    "\n",
    "            c_score.append(20*risk2[row] + 60)\n",
    "\n",
    "        elif c_genotype[row] == gene3[row]:\n",
    "\n",
    "            c_score.append(20*risk3[row] + 80)\n",
    "\n",
    "    print(AG_num + ' finish calculate \"c_score\". ' +'\\n')   \n",
    "\n",
    "\n",
    "\n",
    "    # C result\n",
    "\n",
    "    #############\n",
    "    # (1) 細胞衰老\n",
    "\n",
    "    # 1.細胞循環\n",
    "\n",
    "    g1_1 = mean(c_score[0:5])\n",
    "\n",
    "    # 2.DNA損傷檢查點\n",
    "\n",
    "    g1_2 = mean(c_score[5:10])\n",
    "\n",
    "    # 3.端粒酶活性\n",
    "\n",
    "    g1_3 = mean(c_score[10:12])\n",
    "\n",
    "    # 4.DNA修復\n",
    "\n",
    "    g1_4 = mean(c_score[12:20])\n",
    "\n",
    "    # 5.粒線體活性\n",
    "\n",
    "    g1_5 = mean(c_score[20:26])\n",
    "\n",
    "    # 6.對抗自由基\n",
    "\n",
    "    g1_6 = mean(c_score[26:32])\n",
    "\n",
    "\n",
    "    #############\n",
    "    # (2) 毒物代謝\n",
    "\n",
    "    # 1.肝臟\n",
    "\n",
    "    g2_1 = mean(c_score[32:37])\n",
    "\n",
    "    # 2.肺臟\n",
    "\n",
    "    g2_2 = mean(c_score[37:40])\n",
    "\n",
    "    # 3.腸道\n",
    "\n",
    "    g2_3 = mean(c_score[40:43])\n",
    "\n",
    "    # 4.腎臟\n",
    "\n",
    "    g2_4 = mean(c_score[43:47])\n",
    "\n",
    "\n",
    "    #############\n",
    "    # (3) 營養代謝\n",
    "\n",
    "    # 1.血管健康\n",
    "\n",
    "    g3_1 = mean(c_score[47:51])\n",
    "\n",
    "    # 2.蛋白質代謝\n",
    "\n",
    "    g3_2 = mean(c_score[51:57])\n",
    "\n",
    "    # 3.醣類代謝\n",
    "\n",
    "    g3_3 = mean(c_score[57:65])\n",
    "\n",
    "    # 4.脂質代謝\n",
    "\n",
    "    g3_4 = mean(c_score[65:72])\n",
    "\n",
    "\n",
    "    #############\n",
    "    # (4) 免疫調節\n",
    "\n",
    "    # 1.抗原呈現\n",
    "\n",
    "    g4_1 = mean(c_score[72:77])\n",
    "\n",
    "    # 2.免疫訓練\n",
    "\n",
    "    g4_2 = mean(c_score[77:88])\n",
    "\n",
    "    # 3.發炎因子\n",
    "\n",
    "    g4_3 = mean(c_score[88:97])\n",
    "\n",
    "\n",
    "\n",
    "    #####################\n",
    "    # Score of 4 category\n",
    "\n",
    "    # (1) 細胞衰老\n",
    "\n",
    "    g1 = mean(c_score[0:32])\n",
    "\n",
    "    # (2) 毒物代謝\n",
    "\n",
    "    g2 = mean(c_score[32:47])\n",
    "\n",
    "    # (3) 營養代謝\n",
    "\n",
    "    g3 = mean(c_score[47:72])\n",
    "\n",
    "    # (4) 免疫調節\n",
    "\n",
    "    g4 = mean(c_score[72:97])\n",
    "\n",
    "\n",
    "    #############################################\n",
    "    # Pick the bad subcategory from each category\n",
    "\n",
    "\n",
    "    sub = []\n",
    "\n",
    "    # (1) 細胞衰老\n",
    "\n",
    "    if min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_1:\n",
    "\n",
    "        sub.append(\"細胞循環\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_2:\n",
    "\n",
    "        sub.append(\"DNA損傷檢查點\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_3:\n",
    "\n",
    "        sub.append(\"端粒酶活性\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_4:\n",
    "\n",
    "        sub.append(\"DNA修復\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_5:\n",
    "\n",
    "        sub.append(\"粒線體活性\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_6:\n",
    "\n",
    "        sub.append(\"對抗自由基\")\n",
    "\n",
    "\n",
    "    # (2) 毒物代謝\n",
    "\n",
    "    if min(g2_1, g2_2, g2_3, g2_4) == g2_1:\n",
    "\n",
    "        sub.append(\"肝臟\")\n",
    "\n",
    "    elif min(g2_1, g2_2, g2_3, g2_4) == g2_2:\n",
    "\n",
    "        sub.append(\"肺臟\")\n",
    "\n",
    "    elif min(g2_1, g2_2, g2_3, g2_4) == g2_3:\n",
    "\n",
    "        sub.append(\"腸道\")\n",
    "\n",
    "    elif min(g2_1, g2_2, g2_3, g2_4) == g2_4:\n",
    "\n",
    "        sub.append(\"腎臟\")\n",
    "\n",
    "\n",
    "    # (3) 營養代謝\n",
    "\n",
    "    if min(g3_1, g3_2, g3_3, g3_4) == g3_1:\n",
    "\n",
    "        sub.append(\"血管健康\")\n",
    "\n",
    "    elif min(g3_1, g3_2, g3_3, g3_4) == g3_2:\n",
    "\n",
    "        sub.append(\"蛋白質代謝\")\n",
    "\n",
    "    elif min(g3_1, g3_2, g3_3, g3_4) == g3_3:\n",
    "\n",
    "        sub.append(\"醣類代謝\")\n",
    "\n",
    "    elif min(g3_1, g3_2, g3_3, g3_4) == g3_4:\n",
    "\n",
    "        sub.append(\"脂質代謝\")\n",
    "\n",
    "\n",
    "    # (4) 免疫調節\n",
    "\n",
    "    if min(g4_1, g4_2, g4_3) == g4_1:\n",
    "\n",
    "        sub.append(\"抗原呈現\")\n",
    "\n",
    "    elif min(g4_1, g4_2, g4_3) == g4_2:\n",
    "\n",
    "        sub.append(\"免疫訓練\")\n",
    "\n",
    "    elif min(g4_1, g4_2, g4_3) == g4_3:\n",
    "\n",
    "        sub.append(\"發炎因子\")\n",
    "\n",
    "\n",
    "    print(AG_num + ' finish calculate \"c_result\". ' +'\\n')\n",
    "\n",
    "    sub\n",
    "\n",
    "    #####################\n",
    "    # output c_result.csv\n",
    "\n",
    "    ag_genotype['c_value'] = c_ratio\n",
    "    ag_genotype['c_genotype'] = c_genotype\n",
    "    ag_genotype['c_score'] = c_score\n",
    "\n",
    "    ag_genotype\n",
    "\n",
    "\n",
    "    outname = AG_num + '_c_result.csv'\n",
    "\n",
    "    outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "    if not os.path.exists(outdir):\n",
    "        os.makedirs(outdir)\n",
    "\n",
    "    fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "    ag_genotype.to_csv(fullname)\n",
    "\n",
    "    print(AG_num + ' finish output ' + fullname + '\\n')\n",
    "\n",
    "    \n",
    "    #######################\n",
    "    # output c_category.csv\n",
    "    \n",
    "    category = [\"細胞衰老\", \"毒物代謝\", \"營養代謝\", \"免疫調節\"]\n",
    "\n",
    "    c_mean = [format(g1, '.0f'), format(g2, '.0f'), format(g3, '.0f'), format(g4, '.0f')]\n",
    "\n",
    "    c_category = pd.DataFrame(list(zip(category, c_mean, sub)),\n",
    "                  columns=['category','mean', 'high risk subcategory'])\n",
    "\n",
    "\n",
    "    outname = AG_num + '_c_category.csv'\n",
    "\n",
    "    outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "    if not os.path.exists(outdir):\n",
    "        os.makedirs(outdir)\n",
    "\n",
    "    fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "    c_category.to_csv(fullname)\n",
    "\n",
    "    print(AG_num + ' finish output ' + fullname + '\\n')\n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "elif sample_num.upper() == \"D\":    \n",
    "\n",
    "    # get value (delta vic / delta fam)\n",
    "\n",
    "    d_value = value_vic_fam.iloc[[204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215,\n",
    "                        228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239,\n",
    "                        252, 253, \n",
    "                        239,\n",
    "                        254, 255, 256, 257, 258, 259, 260,\n",
    "                        255, 254,\n",
    "                        261, 262, 263,\n",
    "                        276, 277, 278, 279, 280, 281, 282, 283, 284, 285, 286, 287,\n",
    "                        300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 310, 311,\n",
    "                        324, 325, 326, 327, 328, 329, 330, 331, 332, 333, 334, 335,\n",
    "                        348, 349, 350, 351, 352, 353, 354, 355, 357, 356, 358, 359,\n",
    "                        372, 373, 374, 375, 376, 377, 378, 379, 380, 381], [6]]\n",
    "\n",
    "    d_ratio = []\n",
    "\n",
    "    for row in range(len(d_value)):\n",
    "\n",
    "        d_ratio.append(d_value.iloc[row,0])\n",
    "\n",
    "    print(AG_num + ' finish calculate the \"d_value\".' + '\\n')\n",
    "\n",
    "\n",
    "    # Get genotype\n",
    "\n",
    "    d_genotype = []\n",
    "\n",
    "    for row in range(len(d_value)):\n",
    "\n",
    "        if stand1[row] > stand2[row]:\n",
    "\n",
    "            if float(d_value.iloc[row,]) > stand1[row]:\n",
    "\n",
    "                d_genotype.append(gene1[row])\n",
    "\n",
    "            elif float(d_value.iloc[row,]) < stand2[row]:\n",
    "\n",
    "                d_genotype.append(gene3[row])\n",
    "\n",
    "            else:\n",
    "\n",
    "                d_genotype.append(gene2[row])  \n",
    "\n",
    "        elif stand1[row] < stand2[row]:\n",
    "\n",
    "            if float(d_value.iloc[row,]) > stand2[row]:\n",
    "\n",
    "                d_genotype.append(gene3[row])\n",
    "\n",
    "            elif float(d_value.iloc[row,]) < stand1[row]:\n",
    "\n",
    "                d_genotype.append(gene1[row])\n",
    "\n",
    "            else:\n",
    "\n",
    "                d_genotype.append(gene2[row])  \n",
    "\n",
    "\n",
    "    print(AG_num + ' finish calculate \"d_genotype\".' +'\\n')\n",
    "\n",
    "    #d_genotype\n",
    "\n",
    "\n",
    "\n",
    "    # D score\n",
    "\n",
    "    d_score = []\n",
    "\n",
    "    for row in range(len(d_genotype)):\n",
    "\n",
    "        if d_genotype[row] == gene1[row]:\n",
    "\n",
    "            d_score.append(40*risk1[row] + 20)\n",
    "\n",
    "        elif d_genotype[row] == gene2[row]:\n",
    "\n",
    "            d_score.append(20*risk2[row] + 60)\n",
    "\n",
    "        elif d_genotype[row] == gene3[row]:\n",
    "\n",
    "            d_score.append(20*risk3[row] + 80)\n",
    "\n",
    "    print(AG_num + ' finish calculate \"d_score\". ' +'\\n')   \n",
    "\n",
    "\n",
    "\n",
    "    # D result\n",
    "\n",
    "    #############\n",
    "    # (1) 細胞衰老\n",
    "\n",
    "    # 1.細胞循環\n",
    "\n",
    "    g1_1 = mean(d_score[0:5])\n",
    "\n",
    "    # 2.DNA損傷檢查點\n",
    "\n",
    "    g1_2 = mean(d_score[5:10])\n",
    "\n",
    "    # 3.端粒酶活性\n",
    "\n",
    "    g1_3 = mean(d_score[10:12])\n",
    "\n",
    "    # 4.DNA修復\n",
    "\n",
    "    g1_4 = mean(d_score[12:20])\n",
    "\n",
    "    # 5.粒線體活性\n",
    "\n",
    "    g1_5 = mean(d_score[20:26])\n",
    "\n",
    "    # 6.對抗自由基\n",
    "\n",
    "    g1_6 = mean(d_score[26:32])\n",
    "\n",
    "\n",
    "    #############\n",
    "    # (2) 毒物代謝\n",
    "\n",
    "    # 1.肝臟\n",
    "\n",
    "    g2_1 = mean(d_score[32:37])\n",
    "\n",
    "    # 2.肺臟\n",
    "\n",
    "    g2_2 = mean(d_score[37:40])\n",
    "\n",
    "    # 3.腸道\n",
    "\n",
    "    g2_3 = mean(d_score[40:43])\n",
    "\n",
    "    # 4.腎臟\n",
    "\n",
    "    g2_4 = mean(d_score[43:47])\n",
    "\n",
    "\n",
    "    #############\n",
    "    # (3) 營養代謝\n",
    "\n",
    "    # 1.血管健康\n",
    "\n",
    "    g3_1 = mean(d_score[47:51])\n",
    "\n",
    "    # 2.蛋白質代謝\n",
    "\n",
    "    g3_2 = mean(d_score[51:57])\n",
    "\n",
    "    # 3.醣類代謝\n",
    "\n",
    "    g3_3 = mean(d_score[57:65])\n",
    "\n",
    "    # 4.脂質代謝\n",
    "\n",
    "    g3_4 = mean(d_score[65:72])\n",
    "\n",
    "\n",
    "    #############\n",
    "    # (4) 免疫調節\n",
    "\n",
    "    # 1.抗原呈現\n",
    "\n",
    "    g4_1 = mean(d_score[72:77])\n",
    "\n",
    "    # 2.免疫訓練\n",
    "\n",
    "    g4_2 = mean(d_score[77:88])\n",
    "\n",
    "    # 3.發炎因子\n",
    "\n",
    "    g4_3 = mean(d_score[88:97])\n",
    "\n",
    "\n",
    "\n",
    "    #####################\n",
    "    # Score of 4 category\n",
    "\n",
    "    # (1) 細胞衰老\n",
    "\n",
    "    g1 = mean(d_score[0:32])\n",
    "\n",
    "    # (2) 毒物代謝\n",
    "\n",
    "    g2 = mean(d_score[32:47])\n",
    "\n",
    "    # (3) 營養代謝\n",
    "\n",
    "    g3 = mean(d_score[47:72])\n",
    "\n",
    "    # (4) 免疫調節\n",
    "\n",
    "    g4 = mean(d_score[72:97])\n",
    "\n",
    "\n",
    "    #############################################\n",
    "    # Pick the bad subcategory from each category\n",
    "\n",
    "\n",
    "    sub = []\n",
    "\n",
    "    # (1) 細胞衰老\n",
    "\n",
    "    if min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_1:\n",
    "\n",
    "        sub.append(\"細胞循環\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_2:\n",
    "\n",
    "        sub.append(\"DNA損傷檢查點\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_3:\n",
    "\n",
    "        sub.append(\"端粒酶活性\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_4:\n",
    "\n",
    "        sub.append(\"DNA修復\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_5:\n",
    "\n",
    "        sub.append(\"粒線體活性\")\n",
    "\n",
    "    elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_6:\n",
    "\n",
    "        sub.append(\"對抗自由基\")\n",
    "\n",
    "\n",
    "    # (2) 毒物代謝\n",
    "\n",
    "    if min(g2_1, g2_2, g2_3, g2_4) == g2_1:\n",
    "\n",
    "        sub.append(\"肝臟\")\n",
    "\n",
    "    elif min(g2_1, g2_2, g2_3, g2_4) == g2_2:\n",
    "\n",
    "        sub.append(\"肺臟\")\n",
    "\n",
    "    elif min(g2_1, g2_2, g2_3, g2_4) == g2_3:\n",
    "\n",
    "        sub.append(\"腸道\")\n",
    "\n",
    "    elif min(g2_1, g2_2, g2_3, g2_4) == g2_4:\n",
    "\n",
    "        sub.append(\"腎臟\")\n",
    "\n",
    "\n",
    "    # (3) 營養代謝\n",
    "\n",
    "    if min(g3_1, g3_2, g3_3, g3_4) == g3_1:\n",
    "\n",
    "        sub.append(\"血管健康\")\n",
    "\n",
    "    elif min(g3_1, g3_2, g3_3, g3_4) == g3_2:\n",
    "\n",
    "        sub.append(\"蛋白質代謝\")\n",
    "\n",
    "    elif min(g3_1, g3_2, g3_3, g3_4) == g3_3:\n",
    "\n",
    "        sub.append(\"醣類代謝\")\n",
    "\n",
    "    elif min(g3_1, g3_2, g3_3, g3_4) == g3_4:\n",
    "\n",
    "        sub.append(\"脂質代謝\")\n",
    "\n",
    "\n",
    "    # (4) 免疫調節\n",
    "\n",
    "    if min(g4_1, g4_2, g4_3) == g4_1:\n",
    "\n",
    "        sub.append(\"抗原呈現\")\n",
    "\n",
    "    elif min(g4_1, g4_2, g4_3) == g4_2:\n",
    "\n",
    "        sub.append(\"免疫訓練\")\n",
    "\n",
    "    elif min(g4_1, g4_2, g4_3) == g4_3:\n",
    "\n",
    "        sub.append(\"發炎因子\")\n",
    "\n",
    "\n",
    "    print(AG_num + ' finish calculate \"d_result\". ' +'\\n')\n",
    "\n",
    "    sub\n",
    "\n",
    "    \n",
    "    #####################\n",
    "    # output d_result.csv\n",
    "\n",
    "    ag_genotype['d_value'] = d_ratio\n",
    "    ag_genotype['d_genotype'] = d_genotype\n",
    "    ag_genotype['d_score'] = d_score\n",
    "\n",
    "    ag_genotype\n",
    "\n",
    "\n",
    "    outname = AG_num + '_d_result.csv'\n",
    "\n",
    "    outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "    if not os.path.exists(outdir):\n",
    "        os.makedirs(outdir)\n",
    "\n",
    "    fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "    ag_genotype.to_csv(fullname)\n",
    "\n",
    "    print(AG_num + ' finish output ' + fullname + '\\n')  \n",
    "    \n",
    "    \n",
    "    #######################\n",
    "    # output d_category.csv\n",
    "    \n",
    "    category = [\"細胞衰老\", \"毒物代謝\", \"營養代謝\", \"免疫調節\"]\n",
    "\n",
    "    d_mean = [format(g1, '.0f'), format(g2, '.0f'), format(g3, '.0f'), format(g4, '.0f')]\n",
    "\n",
    "    d_category = pd.DataFrame(list(zip(category, d_mean, sub)),\n",
    "                  columns=['category','mean', 'high risk subcategory'])\n",
    "\n",
    "\n",
    "    outname = AG_num + '_d_category.csv'\n",
    "\n",
    "    outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "    if not os.path.exists(outdir):\n",
    "        os.makedirs(outdir)\n",
    "\n",
    "    fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "    d_category.to_csv(fullname)\n",
    "\n",
    "    print(AG_num + ' finish output ' + fullname + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "---------------------------\n",
      "細胞衰老:  67 分  粒線體活性\n",
      "毒物代謝:  81 分  肝臟\n",
      "營養代謝:  78 分  血管健康\n",
      "免疫調節:  69 分  發炎因子\n",
      "---------------------------\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(\"\")\n",
    "print(\"---------------------------\")\n",
    "print('細胞衰老: ', format(g1, '.0f'), \"分 \", sub[0])\n",
    "print('毒物代謝: ', format(g2, '.0f'), \"分 \", sub[1])\n",
    "print('營養代謝: ', format(g3, '.0f'), \"分 \", sub[2])\n",
    "print('免疫調節: ', format(g4, '.0f'), \"分 \", sub[3])\n",
    "print(\"---------------------------\")\n",
    "print(\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>big-category</th>\n",
       "      <th>small-category</th>\n",
       "      <th>Gene</th>\n",
       "      <th>SNP</th>\n",
       "      <th>Risk Allele(VIC)</th>\n",
       "      <th>Normal allele(FAM)</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Genotype1</th>\n",
       "      <th>Genotype2</th>\n",
       "      <th>Genotype3</th>\n",
       "      <th>Gene+AF8-risk1</th>\n",
       "      <th>Gene+AF8-risk2</th>\n",
       "      <th>Gene+AF8-risk3</th>\n",
       "      <th>Standard1</th>\n",
       "      <th>Standard2</th>\n",
       "      <th>d_value</th>\n",
       "      <th>d_genotype</th>\n",
       "      <th>d_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Cell cycle</td>\n",
       "      <td>CDKN1A</td>\n",
       "      <td>rs1059234</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C/T</td>\n",
       "      <td>CC</td>\n",
       "      <td>CT</td>\n",
       "      <td>TT</td>\n",
       "      <td>0.379</td>\n",
       "      <td>0.330</td>\n",
       "      <td>0.291</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e-01</td>\n",
       "      <td>0.219033</td>\n",
       "      <td>CT</td>\n",
       "      <td>66.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Cell cycle</td>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>rs1333049</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C/G</td>\n",
       "      <td>CC</td>\n",
       "      <td>CG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.214</td>\n",
       "      <td>0.524</td>\n",
       "      <td>0.262</td>\n",
       "      <td>5.000000e-03</td>\n",
       "      <td>2.000000e+00</td>\n",
       "      <td>0.061989</td>\n",
       "      <td>CG</td>\n",
       "      <td>70.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Cell cycle</td>\n",
       "      <td>CCND1</td>\n",
       "      <td>rs9344</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "      <td>GG</td>\n",
       "      <td>GA</td>\n",
       "      <td>AA</td>\n",
       "      <td>0.184</td>\n",
       "      <td>0.505</td>\n",
       "      <td>0.311</td>\n",
       "      <td>1.000000e-03</td>\n",
       "      <td>2.000000e+00</td>\n",
       "      <td>1.178644</td>\n",
       "      <td>GA</td>\n",
       "      <td>70.10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Cell cycle</td>\n",
       "      <td>CDKN1B</td>\n",
       "      <td>rs34330</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T/C</td>\n",
       "      <td>TT</td>\n",
       "      <td>TC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.223</td>\n",
       "      <td>0.534</td>\n",
       "      <td>0.243</td>\n",
       "      <td>1.000000e-01</td>\n",
       "      <td>3.000000e+00</td>\n",
       "      <td>3.157721</td>\n",
       "      <td>CC</td>\n",
       "      <td>84.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Cell cycle</td>\n",
       "      <td>TP53</td>\n",
       "      <td>rs1042522</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G/C</td>\n",
       "      <td>GG</td>\n",
       "      <td>GC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.175</td>\n",
       "      <td>0.553</td>\n",
       "      <td>0.272</td>\n",
       "      <td>3.000000e+00</td>\n",
       "      <td>2.000000e+00</td>\n",
       "      <td>2.959190</td>\n",
       "      <td>GC</td>\n",
       "      <td>71.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Damage Checkpoints</td>\n",
       "      <td>BRCA1</td>\n",
       "      <td>rs799917</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "      <td>GG</td>\n",
       "      <td>GA</td>\n",
       "      <td>AA</td>\n",
       "      <td>0.447</td>\n",
       "      <td>0.408</td>\n",
       "      <td>0.146</td>\n",
       "      <td>3.000000e-01</td>\n",
       "      <td>9.999900e+04</td>\n",
       "      <td>0.162443</td>\n",
       "      <td>GG</td>\n",
       "      <td>37.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Damage Checkpoints</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>rs144848</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>A/C</td>\n",
       "      <td>AA</td>\n",
       "      <td>AC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.573</td>\n",
       "      <td>0.350</td>\n",
       "      <td>0.078</td>\n",
       "      <td>2.000000e+00</td>\n",
       "      <td>2.000000e-01</td>\n",
       "      <td>1.373451</td>\n",
       "      <td>AC</td>\n",
       "      <td>67.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Damage Checkpoints</td>\n",
       "      <td>TP53BP1</td>\n",
       "      <td>rs560191</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C/G</td>\n",
       "      <td>CC</td>\n",
       "      <td>CG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.223</td>\n",
       "      <td>0.485</td>\n",
       "      <td>0.291</td>\n",
       "      <td>1.000000e-03</td>\n",
       "      <td>5.000000e+00</td>\n",
       "      <td>0.370641</td>\n",
       "      <td>CG</td>\n",
       "      <td>69.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Damage Checkpoints</td>\n",
       "      <td>TP73</td>\n",
       "      <td>rs1801173</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T/C</td>\n",
       "      <td>TT</td>\n",
       "      <td>TC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.058</td>\n",
       "      <td>0.408</td>\n",
       "      <td>0.534</td>\n",
       "      <td>1.000000e-05</td>\n",
       "      <td>5.000000e-01</td>\n",
       "      <td>7.049019</td>\n",
       "      <td>CC</td>\n",
       "      <td>90.68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Damage Checkpoints</td>\n",
       "      <td>RAD51</td>\n",
       "      <td>rs1801320</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C/G</td>\n",
       "      <td>CC</td>\n",
       "      <td>CG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.291</td>\n",
       "      <td>0.709</td>\n",
       "      <td>1.000000e-05</td>\n",
       "      <td>5.000000e-01</td>\n",
       "      <td>3.126596</td>\n",
       "      <td>GG</td>\n",
       "      <td>94.18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Telomerase Activity</td>\n",
       "      <td>TERT</td>\n",
       "      <td>rs2736100</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>A/C</td>\n",
       "      <td>AA</td>\n",
       "      <td>AC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.262</td>\n",
       "      <td>0.631</td>\n",
       "      <td>0.107</td>\n",
       "      <td>2.500000e+00</td>\n",
       "      <td>6.000000e-01</td>\n",
       "      <td>0.134487</td>\n",
       "      <td>CC</td>\n",
       "      <td>82.14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Telomerase Activity</td>\n",
       "      <td>TERC</td>\n",
       "      <td>rs12696304</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G/C</td>\n",
       "      <td>GG</td>\n",
       "      <td>GC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.515</td>\n",
       "      <td>0.427</td>\n",
       "      <td>0.058</td>\n",
       "      <td>1.500000e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.106707</td>\n",
       "      <td>GG</td>\n",
       "      <td>40.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Repair</td>\n",
       "      <td>ERCC1</td>\n",
       "      <td>rs2298881</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>A/C</td>\n",
       "      <td>AA</td>\n",
       "      <td>AC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.223</td>\n",
       "      <td>0.408</td>\n",
       "      <td>0.369</td>\n",
       "      <td>4.800000e+00</td>\n",
       "      <td>5.000000e+00</td>\n",
       "      <td>5.030878</td>\n",
       "      <td>CC</td>\n",
       "      <td>87.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Repair</td>\n",
       "      <td>XRCC1</td>\n",
       "      <td>rs25487</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C/T</td>\n",
       "      <td>CC</td>\n",
       "      <td>CT</td>\n",
       "      <td>TT</td>\n",
       "      <td>0.553</td>\n",
       "      <td>0.388</td>\n",
       "      <td>0.058</td>\n",
       "      <td>1.000000e-03</td>\n",
       "      <td>1.000000e-04</td>\n",
       "      <td>2.149230</td>\n",
       "      <td>CC</td>\n",
       "      <td>42.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Repair</td>\n",
       "      <td>XRCC4</td>\n",
       "      <td>rs1805377</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.466</td>\n",
       "      <td>0.427</td>\n",
       "      <td>0.107</td>\n",
       "      <td>7.000000e-01</td>\n",
       "      <td>4.500000e-01</td>\n",
       "      <td>0.732262</td>\n",
       "      <td>AA</td>\n",
       "      <td>38.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Repair</td>\n",
       "      <td>ATM</td>\n",
       "      <td>rs1801516</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.029</td>\n",
       "      <td>0.971</td>\n",
       "      <td>9.999900e+04</td>\n",
       "      <td>3.000000e-01</td>\n",
       "      <td>0.137206</td>\n",
       "      <td>GG</td>\n",
       "      <td>99.42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Repair</td>\n",
       "      <td>PARP1</td>\n",
       "      <td>rs1136410</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.165</td>\n",
       "      <td>0.524</td>\n",
       "      <td>0.311</td>\n",
       "      <td>5.200000e-01</td>\n",
       "      <td>4.000000e-01</td>\n",
       "      <td>0.364818</td>\n",
       "      <td>GG</td>\n",
       "      <td>86.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Repair</td>\n",
       "      <td>FEN1</td>\n",
       "      <td>rs174538</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "      <td>GG</td>\n",
       "      <td>GA</td>\n",
       "      <td>AA</td>\n",
       "      <td>0.427</td>\n",
       "      <td>0.447</td>\n",
       "      <td>0.126</td>\n",
       "      <td>3.000000e-01</td>\n",
       "      <td>4.000000e-01</td>\n",
       "      <td>0.159425</td>\n",
       "      <td>GG</td>\n",
       "      <td>37.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Repair</td>\n",
       "      <td>MRE11A</td>\n",
       "      <td>rs13447720</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C/T</td>\n",
       "      <td>CC</td>\n",
       "      <td>CT</td>\n",
       "      <td>TT</td>\n",
       "      <td>0.049</td>\n",
       "      <td>0.301</td>\n",
       "      <td>0.650</td>\n",
       "      <td>9.999900e+04</td>\n",
       "      <td>1.400000e-01</td>\n",
       "      <td>0.138628</td>\n",
       "      <td>TT</td>\n",
       "      <td>93.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Cell</td>\n",
       "      <td>DNA Repair</td>\n",
       "      <td>hOGG1</td>\n",
       "      <td>rs1052133</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G/C</td>\n",
       "      <td>GG</td>\n",
       "      <td>GC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.282</td>\n",
       "      <td>0.553</td>\n",
       "      <td>0.165</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e-01</td>\n",
       "      <td>0.272667</td>\n",
       "      <td>GC</td>\n",
       "      <td>71.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Mitochondrial Activity</td>\n",
       "      <td>FOXO1</td>\n",
       "      <td>rs2755213</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T/C</td>\n",
       "      <td>TT</td>\n",
       "      <td>TC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.075</td>\n",
       "      <td>0.476</td>\n",
       "      <td>0.320</td>\n",
       "      <td>5.000000e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.096738</td>\n",
       "      <td>TT</td>\n",
       "      <td>23.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Mitochondrial Activity</td>\n",
       "      <td>FOXO3a</td>\n",
       "      <td>rs2802292</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>T/G</td>\n",
       "      <td>TT</td>\n",
       "      <td>TG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.670</td>\n",
       "      <td>0.282</td>\n",
       "      <td>0.049</td>\n",
       "      <td>1.000000e-01</td>\n",
       "      <td>6.000000e-03</td>\n",
       "      <td>0.065999</td>\n",
       "      <td>TG</td>\n",
       "      <td>65.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Mitochondrial Activity</td>\n",
       "      <td>SIRT1</td>\n",
       "      <td>rs3758391</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T/C</td>\n",
       "      <td>TT</td>\n",
       "      <td>TC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.660</td>\n",
       "      <td>0.330</td>\n",
       "      <td>0.010</td>\n",
       "      <td>5.000000e-01</td>\n",
       "      <td>6.000000e-01</td>\n",
       "      <td>0.027877</td>\n",
       "      <td>TT</td>\n",
       "      <td>46.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Mitochondrial Activity</td>\n",
       "      <td>SOD2</td>\n",
       "      <td>rs4880</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "      <td>GG</td>\n",
       "      <td>GA</td>\n",
       "      <td>AA</td>\n",
       "      <td>0.010</td>\n",
       "      <td>0.214</td>\n",
       "      <td>0.777</td>\n",
       "      <td>1.000000e-05</td>\n",
       "      <td>8.000000e-01</td>\n",
       "      <td>0.761502</td>\n",
       "      <td>GA</td>\n",
       "      <td>64.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Mitochondrial Activity</td>\n",
       "      <td>UCP3</td>\n",
       "      <td>rs15763</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "      <td>GG</td>\n",
       "      <td>GA</td>\n",
       "      <td>AA</td>\n",
       "      <td>0.757</td>\n",
       "      <td>0.214</td>\n",
       "      <td>0.029</td>\n",
       "      <td>1.500000e-01</td>\n",
       "      <td>9.999900e+04</td>\n",
       "      <td>0.118210</td>\n",
       "      <td>GG</td>\n",
       "      <td>50.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Mitochondrial Activity</td>\n",
       "      <td>UCP2</td>\n",
       "      <td>rs660339</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.252</td>\n",
       "      <td>0.456</td>\n",
       "      <td>0.291</td>\n",
       "      <td>5.000000e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.466374</td>\n",
       "      <td>AA</td>\n",
       "      <td>30.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Against Free Radicals</td>\n",
       "      <td>SOD2</td>\n",
       "      <td>rs4880</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "      <td>GG</td>\n",
       "      <td>GA</td>\n",
       "      <td>AA</td>\n",
       "      <td>0.010</td>\n",
       "      <td>0.214</td>\n",
       "      <td>0.777</td>\n",
       "      <td>1.000000e-05</td>\n",
       "      <td>8.000000e-01</td>\n",
       "      <td>0.761502</td>\n",
       "      <td>GA</td>\n",
       "      <td>64.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Against Free Radicals</td>\n",
       "      <td>GSTP1</td>\n",
       "      <td>rs1695</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "      <td>GG</td>\n",
       "      <td>GA</td>\n",
       "      <td>AA</td>\n",
       "      <td>0.019</td>\n",
       "      <td>0.330</td>\n",
       "      <td>0.650</td>\n",
       "      <td>1.000000e-01</td>\n",
       "      <td>2.000000e-01</td>\n",
       "      <td>1.584336</td>\n",
       "      <td>AA</td>\n",
       "      <td>93.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Against Free Radicals</td>\n",
       "      <td>GSTM1</td>\n",
       "      <td>rs366631</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.500</td>\n",
       "      <td>0.500</td>\n",
       "      <td>0.999</td>\n",
       "      <td>5.000000e-01</td>\n",
       "      <td>2.000000e+00</td>\n",
       "      <td>1.356859</td>\n",
       "      <td>AG</td>\n",
       "      <td>70.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Cell</td>\n",
       "      <td>Against Free Radicals</td>\n",
       "      <td>GPX1</td>\n",
       "      <td>rs1800668</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.175</td>\n",
       "      <td>0.825</td>\n",
       "      <td>1.000000e-05</td>\n",
       "      <td>2.000000e+00</td>\n",
       "      <td>2.583307</td>\n",
       "      <td>GG</td>\n",
       "      <td>96.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>Nutrients</td>\n",
       "      <td>Fat Metabolism</td>\n",
       "      <td>APOB</td>\n",
       "      <td>rs693</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.107</td>\n",
       "      <td>0.893</td>\n",
       "      <td>9.999900e+04</td>\n",
       "      <td>4.000000e-01</td>\n",
       "      <td>0.097417</td>\n",
       "      <td>GG</td>\n",
       "      <td>97.86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>Nutrients</td>\n",
       "      <td>Fat Metabolism</td>\n",
       "      <td>APOC3</td>\n",
       "      <td>rs4520</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C/T</td>\n",
       "      <td>CC</td>\n",
       "      <td>CT</td>\n",
       "      <td>TT</td>\n",
       "      <td>0.146</td>\n",
       "      <td>0.485</td>\n",
       "      <td>0.369</td>\n",
       "      <td>9.999900e+04</td>\n",
       "      <td>2.500000e-01</td>\n",
       "      <td>0.009986</td>\n",
       "      <td>TT</td>\n",
       "      <td>87.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>Nutrients</td>\n",
       "      <td>Fat Metabolism</td>\n",
       "      <td>APOA5</td>\n",
       "      <td>rs651821</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C/T</td>\n",
       "      <td>CC</td>\n",
       "      <td>CT</td>\n",
       "      <td>TT</td>\n",
       "      <td>0.058</td>\n",
       "      <td>0.398</td>\n",
       "      <td>0.544</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>2.000000e-01</td>\n",
       "      <td>0.432137</td>\n",
       "      <td>CT</td>\n",
       "      <td>67.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>Nutrients</td>\n",
       "      <td>Fat Metabolism</td>\n",
       "      <td>CETP</td>\n",
       "      <td>rs2303790</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "      <td>GG</td>\n",
       "      <td>GA</td>\n",
       "      <td>AA</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.049</td>\n",
       "      <td>0.951</td>\n",
       "      <td>1.000000e-05</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.560999</td>\n",
       "      <td>AA</td>\n",
       "      <td>99.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>Nutrients</td>\n",
       "      <td>Fat Metabolism</td>\n",
       "      <td>LPL</td>\n",
       "      <td>rs328</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>G/C</td>\n",
       "      <td>GG</td>\n",
       "      <td>GC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.204</td>\n",
       "      <td>0.796</td>\n",
       "      <td>1.000000e-05</td>\n",
       "      <td>8.000000e-01</td>\n",
       "      <td>0.950663</td>\n",
       "      <td>CC</td>\n",
       "      <td>95.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Antigen Presentation</td>\n",
       "      <td>TLR+AC0-1</td>\n",
       "      <td>rs5743551</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C/T</td>\n",
       "      <td>CC</td>\n",
       "      <td>CT</td>\n",
       "      <td>TT</td>\n",
       "      <td>0.456</td>\n",
       "      <td>0.466</td>\n",
       "      <td>0.078</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>4.000000e-01</td>\n",
       "      <td>0.448746</td>\n",
       "      <td>CT</td>\n",
       "      <td>69.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Antigen Presentation</td>\n",
       "      <td>TLR+AC0-3</td>\n",
       "      <td>rs3775291</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T/C</td>\n",
       "      <td>TT</td>\n",
       "      <td>TC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.107</td>\n",
       "      <td>0.369</td>\n",
       "      <td>0.524</td>\n",
       "      <td>5.000000e-02</td>\n",
       "      <td>4.500000e-02</td>\n",
       "      <td>0.760001</td>\n",
       "      <td>TT</td>\n",
       "      <td>24.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Antigen Presentation</td>\n",
       "      <td>TLR+AC0-8</td>\n",
       "      <td>rs3764879</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C/G</td>\n",
       "      <td>CC</td>\n",
       "      <td>CG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.233</td>\n",
       "      <td>0.320</td>\n",
       "      <td>9.999900e+04</td>\n",
       "      <td>3.000000e-01</td>\n",
       "      <td>0.087483</td>\n",
       "      <td>GG</td>\n",
       "      <td>86.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Antigen Presentation</td>\n",
       "      <td>TLR+AC0-9</td>\n",
       "      <td>rs5743836</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "      <td>GG</td>\n",
       "      <td>GA</td>\n",
       "      <td>AA</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0.019</td>\n",
       "      <td>0.999</td>\n",
       "      <td>5.000000e-01</td>\n",
       "      <td>7.000000e-01</td>\n",
       "      <td>1.998541</td>\n",
       "      <td>AA</td>\n",
       "      <td>99.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Antigen Presentation</td>\n",
       "      <td>HLA</td>\n",
       "      <td>rs2860580</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.126</td>\n",
       "      <td>0.515</td>\n",
       "      <td>0.359</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e-01</td>\n",
       "      <td>0.383795</td>\n",
       "      <td>AG</td>\n",
       "      <td>70.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Genes Involved in Adaptive Immune System</td>\n",
       "      <td>IL12A</td>\n",
       "      <td>rs583911</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.068</td>\n",
       "      <td>0.408</td>\n",
       "      <td>0.524</td>\n",
       "      <td>1.000000e-04</td>\n",
       "      <td>6.000000e-01</td>\n",
       "      <td>0.575595</td>\n",
       "      <td>AG</td>\n",
       "      <td>68.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Genes Involved in Adaptive Immune System</td>\n",
       "      <td>IL12B</td>\n",
       "      <td>rs3181224</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "      <td>GG</td>\n",
       "      <td>GA</td>\n",
       "      <td>AA</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.087</td>\n",
       "      <td>0.913</td>\n",
       "      <td>1.000000e-01</td>\n",
       "      <td>2.000000e-01</td>\n",
       "      <td>2.409493</td>\n",
       "      <td>AA</td>\n",
       "      <td>98.26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Genes Involved in Adaptive Immune System</td>\n",
       "      <td>STAT4</td>\n",
       "      <td>rs16833215</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>G/A</td>\n",
       "      <td>GG</td>\n",
       "      <td>GA</td>\n",
       "      <td>AA</td>\n",
       "      <td>0.184</td>\n",
       "      <td>0.505</td>\n",
       "      <td>0.311</td>\n",
       "      <td>3.500000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>3.503678</td>\n",
       "      <td>GG</td>\n",
       "      <td>27.36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Genes Involved in Adaptive Immune System</td>\n",
       "      <td>PHF11</td>\n",
       "      <td>rs1925742</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C/T</td>\n",
       "      <td>CC</td>\n",
       "      <td>CT</td>\n",
       "      <td>TT</td>\n",
       "      <td>0.272</td>\n",
       "      <td>0.485</td>\n",
       "      <td>0.243</td>\n",
       "      <td>6.450000e-01</td>\n",
       "      <td>5.000000e-01</td>\n",
       "      <td>0.647525</td>\n",
       "      <td>CC</td>\n",
       "      <td>30.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Genes Involved in Adaptive Immune System</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>rs3781093</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C/T</td>\n",
       "      <td>CC</td>\n",
       "      <td>CT</td>\n",
       "      <td>TT</td>\n",
       "      <td>0.049</td>\n",
       "      <td>0.466</td>\n",
       "      <td>0.505</td>\n",
       "      <td>1.500000e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.213565</td>\n",
       "      <td>CT</td>\n",
       "      <td>69.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Genes Involved in Adaptive Immune System</td>\n",
       "      <td>STAT6</td>\n",
       "      <td>rs1059513</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C/T</td>\n",
       "      <td>CC</td>\n",
       "      <td>CT</td>\n",
       "      <td>TT</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.136</td>\n",
       "      <td>0.864</td>\n",
       "      <td>1.300000e+00</td>\n",
       "      <td>7.000000e-01</td>\n",
       "      <td>0.073876</td>\n",
       "      <td>TT</td>\n",
       "      <td>97.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Genes Involved in Adaptive Immune System</td>\n",
       "      <td>CTLA4</td>\n",
       "      <td>rs926169</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>T/G</td>\n",
       "      <td>TT</td>\n",
       "      <td>TG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.466</td>\n",
       "      <td>0.427</td>\n",
       "      <td>0.107</td>\n",
       "      <td>9.000000e-01</td>\n",
       "      <td>1.000000e-06</td>\n",
       "      <td>0.220932</td>\n",
       "      <td>TG</td>\n",
       "      <td>68.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Genes Involved in Adaptive Immune System</td>\n",
       "      <td>IL10</td>\n",
       "      <td>rs1554286</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.563</td>\n",
       "      <td>0.340</td>\n",
       "      <td>0.097</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>2.000000e-01</td>\n",
       "      <td>2.034827</td>\n",
       "      <td>AA</td>\n",
       "      <td>42.52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Genes Involved in Adaptive Immune System</td>\n",
       "      <td>IL12RA</td>\n",
       "      <td>rs706778</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T/C</td>\n",
       "      <td>TT</td>\n",
       "      <td>TC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.379</td>\n",
       "      <td>0.485</td>\n",
       "      <td>0.136</td>\n",
       "      <td>6.000000e-02</td>\n",
       "      <td>1.000000e-05</td>\n",
       "      <td>0.057823</td>\n",
       "      <td>TC</td>\n",
       "      <td>69.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Genes Involved in Adaptive Immune System</td>\n",
       "      <td>MME</td>\n",
       "      <td>rs1385477</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>T/G</td>\n",
       "      <td>TT</td>\n",
       "      <td>TG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.058</td>\n",
       "      <td>0.330</td>\n",
       "      <td>0.612</td>\n",
       "      <td>2.000000e-01</td>\n",
       "      <td>1.200000e+00</td>\n",
       "      <td>1.354940</td>\n",
       "      <td>GG</td>\n",
       "      <td>92.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Genes Involved in Adaptive Immune System</td>\n",
       "      <td>TNFA</td>\n",
       "      <td>rs361525</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.010</td>\n",
       "      <td>0.049</td>\n",
       "      <td>0.942</td>\n",
       "      <td>9.999900e+04</td>\n",
       "      <td>2.000000e+04</td>\n",
       "      <td>0.088828</td>\n",
       "      <td>GG</td>\n",
       "      <td>98.84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Inflammatory Cytokines</td>\n",
       "      <td>CRP</td>\n",
       "      <td>rs1205</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C/T</td>\n",
       "      <td>CC</td>\n",
       "      <td>CT</td>\n",
       "      <td>TT</td>\n",
       "      <td>0.194</td>\n",
       "      <td>0.515</td>\n",
       "      <td>0.291</td>\n",
       "      <td>9.999999e+06</td>\n",
       "      <td>4.000000e-01</td>\n",
       "      <td>0.379395</td>\n",
       "      <td>TT</td>\n",
       "      <td>85.82</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Inflammatory Cytokines</td>\n",
       "      <td>TGFB</td>\n",
       "      <td>rs1800469</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.233</td>\n",
       "      <td>0.485</td>\n",
       "      <td>0.282</td>\n",
       "      <td>3.000000e-01</td>\n",
       "      <td>9.999990e+05</td>\n",
       "      <td>0.323409</td>\n",
       "      <td>AG</td>\n",
       "      <td>69.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Inflammatory Cytokines</td>\n",
       "      <td>IL6</td>\n",
       "      <td>rs1800796</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>C/G</td>\n",
       "      <td>CC</td>\n",
       "      <td>CG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.505</td>\n",
       "      <td>0.427</td>\n",
       "      <td>0.068</td>\n",
       "      <td>3.000000e+00</td>\n",
       "      <td>9.999999e+06</td>\n",
       "      <td>1.674857</td>\n",
       "      <td>CC</td>\n",
       "      <td>40.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Inflammatory Cytokines</td>\n",
       "      <td>IL17A</td>\n",
       "      <td>rs2275913</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.233</td>\n",
       "      <td>0.515</td>\n",
       "      <td>0.252</td>\n",
       "      <td>9.999999e+06</td>\n",
       "      <td>4.000000e-02</td>\n",
       "      <td>0.043170</td>\n",
       "      <td>AG</td>\n",
       "      <td>70.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Inflammatory Cytokines</td>\n",
       "      <td>IL1A</td>\n",
       "      <td>rs17561</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>C/A</td>\n",
       "      <td>CC</td>\n",
       "      <td>CA</td>\n",
       "      <td>AA</td>\n",
       "      <td>0.874</td>\n",
       "      <td>0.126</td>\n",
       "      <td>0.999</td>\n",
       "      <td>1.000000e-01</td>\n",
       "      <td>9.999900e+04</td>\n",
       "      <td>0.034762</td>\n",
       "      <td>CC</td>\n",
       "      <td>54.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Inflammatory Cytokines</td>\n",
       "      <td>IL1B</td>\n",
       "      <td>rs1143634</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>A/G</td>\n",
       "      <td>AA</td>\n",
       "      <td>AG</td>\n",
       "      <td>GG</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.039</td>\n",
       "      <td>0.961</td>\n",
       "      <td>9.999999e+06</td>\n",
       "      <td>7.000000e+00</td>\n",
       "      <td>3.099021</td>\n",
       "      <td>GG</td>\n",
       "      <td>99.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Inflammatory Cytokines</td>\n",
       "      <td>IL8</td>\n",
       "      <td>rs4073</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>A/T</td>\n",
       "      <td>AA</td>\n",
       "      <td>AT</td>\n",
       "      <td>TT</td>\n",
       "      <td>0.184</td>\n",
       "      <td>0.466</td>\n",
       "      <td>0.350</td>\n",
       "      <td>1.000000e-07</td>\n",
       "      <td>3.000000e+00</td>\n",
       "      <td>0.456020</td>\n",
       "      <td>AT</td>\n",
       "      <td>69.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Inflammatory Cytokines</td>\n",
       "      <td>IL4</td>\n",
       "      <td>rs2243250</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>T/C</td>\n",
       "      <td>TT</td>\n",
       "      <td>TC</td>\n",
       "      <td>CC</td>\n",
       "      <td>0.515</td>\n",
       "      <td>0.447</td>\n",
       "      <td>0.039</td>\n",
       "      <td>5.000000e+00</td>\n",
       "      <td>9.999900e+04</td>\n",
       "      <td>0.162091</td>\n",
       "      <td>TT</td>\n",
       "      <td>40.60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>Immune response</td>\n",
       "      <td>Inflammatory Cytokines</td>\n",
       "      <td>PTGS2</td>\n",
       "      <td>rs689466</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>C/T</td>\n",
       "      <td>CC</td>\n",
       "      <td>CT</td>\n",
       "      <td>TT</td>\n",
       "      <td>0.214</td>\n",
       "      <td>0.515</td>\n",
       "      <td>0.272</td>\n",
       "      <td>1.000000e-06</td>\n",
       "      <td>5.000000e-01</td>\n",
       "      <td>0.114713</td>\n",
       "      <td>CT</td>\n",
       "      <td>70.30</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>97 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       big-category                            small-category       Gene  \\\n",
       "0              Cell                                Cell cycle     CDKN1A   \n",
       "1              Cell                                Cell cycle     CDKN2A   \n",
       "2              Cell                                Cell cycle      CCND1   \n",
       "3              Cell                                Cell cycle     CDKN1B   \n",
       "4              Cell                                Cell cycle       TP53   \n",
       "5              Cell                    DNA Damage Checkpoints      BRCA1   \n",
       "6              Cell                    DNA Damage Checkpoints      BRCA2   \n",
       "7              Cell                    DNA Damage Checkpoints    TP53BP1   \n",
       "8              Cell                    DNA Damage Checkpoints       TP73   \n",
       "9              Cell                    DNA Damage Checkpoints      RAD51   \n",
       "10             Cell                       Telomerase Activity       TERT   \n",
       "11             Cell                       Telomerase Activity       TERC   \n",
       "12             Cell                                DNA Repair      ERCC1   \n",
       "13             Cell                                DNA Repair      XRCC1   \n",
       "14             Cell                                DNA Repair      XRCC4   \n",
       "15             Cell                                DNA Repair        ATM   \n",
       "16             Cell                                DNA Repair      PARP1   \n",
       "17             Cell                                DNA Repair       FEN1   \n",
       "18             Cell                                DNA Repair     MRE11A   \n",
       "19             Cell                                DNA Repair      hOGG1   \n",
       "20             Cell                    Mitochondrial Activity      FOXO1   \n",
       "21             Cell                    Mitochondrial Activity     FOXO3a   \n",
       "22             Cell                    Mitochondrial Activity      SIRT1   \n",
       "23             Cell                    Mitochondrial Activity       SOD2   \n",
       "24             Cell                    Mitochondrial Activity       UCP3   \n",
       "25             Cell                    Mitochondrial Activity       UCP2   \n",
       "26             Cell                     Against Free Radicals       SOD2   \n",
       "27             Cell                     Against Free Radicals      GSTP1   \n",
       "28             Cell                     Against Free Radicals      GSTM1   \n",
       "29             Cell                     Against Free Radicals       GPX1   \n",
       "..              ...                                       ...        ...   \n",
       "67        Nutrients                            Fat Metabolism       APOB   \n",
       "68        Nutrients                            Fat Metabolism      APOC3   \n",
       "69        Nutrients                            Fat Metabolism      APOA5   \n",
       "70        Nutrients                            Fat Metabolism       CETP   \n",
       "71        Nutrients                            Fat Metabolism        LPL   \n",
       "72  Immune response                      Antigen Presentation  TLR+AC0-1   \n",
       "73  Immune response                      Antigen Presentation  TLR+AC0-3   \n",
       "74  Immune response                      Antigen Presentation  TLR+AC0-8   \n",
       "75  Immune response                      Antigen Presentation  TLR+AC0-9   \n",
       "76  Immune response                      Antigen Presentation        HLA   \n",
       "77  Immune response  Genes Involved in Adaptive Immune System      IL12A   \n",
       "78  Immune response  Genes Involved in Adaptive Immune System      IL12B   \n",
       "79  Immune response  Genes Involved in Adaptive Immune System      STAT4   \n",
       "80  Immune response  Genes Involved in Adaptive Immune System      PHF11   \n",
       "81  Immune response  Genes Involved in Adaptive Immune System      GATA3   \n",
       "82  Immune response  Genes Involved in Adaptive Immune System      STAT6   \n",
       "83  Immune response  Genes Involved in Adaptive Immune System      CTLA4   \n",
       "84  Immune response  Genes Involved in Adaptive Immune System       IL10   \n",
       "85  Immune response  Genes Involved in Adaptive Immune System     IL12RA   \n",
       "86  Immune response  Genes Involved in Adaptive Immune System        MME   \n",
       "87  Immune response  Genes Involved in Adaptive Immune System       TNFA   \n",
       "88  Immune response                    Inflammatory Cytokines        CRP   \n",
       "89  Immune response                    Inflammatory Cytokines       TGFB   \n",
       "90  Immune response                    Inflammatory Cytokines        IL6   \n",
       "91  Immune response                    Inflammatory Cytokines      IL17A   \n",
       "92  Immune response                    Inflammatory Cytokines       IL1A   \n",
       "93  Immune response                    Inflammatory Cytokines       IL1B   \n",
       "94  Immune response                    Inflammatory Cytokines        IL8   \n",
       "95  Immune response                    Inflammatory Cytokines        IL4   \n",
       "96  Immune response                    Inflammatory Cytokines      PTGS2   \n",
       "\n",
       "           SNP Risk Allele(VIC) Normal allele(FAM) Genotype Genotype1  \\\n",
       "0    rs1059234                C                  T      C/T        CC   \n",
       "1    rs1333049                C                  G      C/G        CC   \n",
       "2       rs9344                G                  A      G/A        GG   \n",
       "3      rs34330                T                  C      T/C        TT   \n",
       "4    rs1042522                G                  C      G/C        GG   \n",
       "5     rs799917                G                  A      G/A        GG   \n",
       "6     rs144848                A                  C      A/C        AA   \n",
       "7     rs560191                C                  G      C/G        CC   \n",
       "8    rs1801173                T                  C      T/C        TT   \n",
       "9    rs1801320                C                  G      C/G        CC   \n",
       "10   rs2736100                A                  C      A/C        AA   \n",
       "11  rs12696304                G                  C      G/C        GG   \n",
       "12   rs2298881                A                  C      A/C        AA   \n",
       "13     rs25487                C                  T      C/T        CC   \n",
       "14   rs1805377                A                  G      A/G        AA   \n",
       "15   rs1801516                A                  G      A/G        AA   \n",
       "16   rs1136410                A                  G      A/G        AA   \n",
       "17    rs174538                G                  A      G/A        GG   \n",
       "18  rs13447720                C                  T      C/T        CC   \n",
       "19   rs1052133                G                  C      G/C        GG   \n",
       "20   rs2755213                T                  C      T/C        TT   \n",
       "21   rs2802292                T                  G      T/G        TT   \n",
       "22   rs3758391                T                  C      T/C        TT   \n",
       "23      rs4880                G                  A      G/A        GG   \n",
       "24     rs15763                G                  A      G/A        GG   \n",
       "25    rs660339                A                  G      A/G        AA   \n",
       "26      rs4880                G                  A      G/A        GG   \n",
       "27     rs1695                 G                  A      G/A        GG   \n",
       "28    rs366631                A                  G      A/G        AA   \n",
       "29   rs1800668                A                  G      A/G        AA   \n",
       "..         ...              ...                ...      ...       ...   \n",
       "67       rs693                A                  G      A/G        AA   \n",
       "68      rs4520                C                  T      C/T        CC   \n",
       "69    rs651821                C                  T      C/T        CC   \n",
       "70   rs2303790                G                  A      G/A        GG   \n",
       "71       rs328                G                  C      G/C        GG   \n",
       "72   rs5743551                C                  T      C/T        CC   \n",
       "73   rs3775291                T                  C      T/C        TT   \n",
       "74   rs3764879                C                  G      C/G        CC   \n",
       "75   rs5743836                G                  A      G/A        GG   \n",
       "76   rs2860580                A                  G      A/G        AA   \n",
       "77    rs583911                A                  G      A/G        AA   \n",
       "78   rs3181224                G                  A      G/A        GG   \n",
       "79  rs16833215                G                  A      G/A        GG   \n",
       "80   rs1925742                C                  T      C/T        CC   \n",
       "81   rs3781093                C                  T      C/T        CC   \n",
       "82   rs1059513                C                  T      C/T        CC   \n",
       "83    rs926169                T                  G      T/G        TT   \n",
       "84   rs1554286                A                  G      A/G        AA   \n",
       "85    rs706778                T                  C      T/C        TT   \n",
       "86   rs1385477                T                  G      T/G        TT   \n",
       "87    rs361525                A                  G      A/G        AA   \n",
       "88      rs1205                C                  T      C/T        CC   \n",
       "89   rs1800469                A                  G      A/G        AA   \n",
       "90   rs1800796                C                  G      C/G        CC   \n",
       "91   rs2275913                A                  G      A/G        AA   \n",
       "92     rs17561                C                  A      C/A        CC   \n",
       "93   rs1143634                A                  G      A/G        AA   \n",
       "94      rs4073                A                  T      A/T        AA   \n",
       "95   rs2243250                T                  C      T/C        TT   \n",
       "96    rs689466                C                  T      C/T        CC   \n",
       "\n",
       "   Genotype2 Genotype3  Gene+AF8-risk1  Gene+AF8-risk2  Gene+AF8-risk3  \\\n",
       "0         CT        TT           0.379           0.330           0.291   \n",
       "1         CG        GG           0.214           0.524           0.262   \n",
       "2         GA        AA           0.184           0.505           0.311   \n",
       "3         TC        CC           0.223           0.534           0.243   \n",
       "4         GC        CC           0.175           0.553           0.272   \n",
       "5         GA        AA           0.447           0.408           0.146   \n",
       "6         AC        CC           0.573           0.350           0.078   \n",
       "7         CG        GG           0.223           0.485           0.291   \n",
       "8         TC        CC           0.058           0.408           0.534   \n",
       "9         CG        GG           0.001           0.291           0.709   \n",
       "10        AC        CC           0.262           0.631           0.107   \n",
       "11        GC        CC           0.515           0.427           0.058   \n",
       "12        AC        CC           0.223           0.408           0.369   \n",
       "13        CT        TT           0.553           0.388           0.058   \n",
       "14        AG        GG           0.466           0.427           0.107   \n",
       "15        AG        GG           0.001           0.029           0.971   \n",
       "16        AG        GG           0.165           0.524           0.311   \n",
       "17        GA        AA           0.427           0.447           0.126   \n",
       "18        CT        TT           0.049           0.301           0.650   \n",
       "19        GC        CC           0.282           0.553           0.165   \n",
       "20        TC        CC           0.075           0.476           0.320   \n",
       "21        TG        GG           0.670           0.282           0.049   \n",
       "22        TC        CC           0.660           0.330           0.010   \n",
       "23        GA        AA           0.010           0.214           0.777   \n",
       "24        GA        AA           0.757           0.214           0.029   \n",
       "25        AG        GG           0.252           0.456           0.291   \n",
       "26        GA        AA           0.010           0.214           0.777   \n",
       "27        GA        AA           0.019           0.330           0.650   \n",
       "28        AG        GG           0.500           0.500           0.999   \n",
       "29        AG        GG           0.001           0.175           0.825   \n",
       "..       ...       ...             ...             ...             ...   \n",
       "67        AG        GG           0.001           0.107           0.893   \n",
       "68        CT        TT           0.146           0.485           0.369   \n",
       "69        CT        TT           0.058           0.398           0.544   \n",
       "70        GA        AA           0.001           0.049           0.951   \n",
       "71        GC        CC           0.001           0.204           0.796   \n",
       "72        CT        TT           0.456           0.466           0.078   \n",
       "73        TC        CC           0.107           0.369           0.524   \n",
       "74        CG        GG           0.001           0.233           0.320   \n",
       "75        GA        AA           0.981           0.019           0.999   \n",
       "76        AG        GG           0.126           0.515           0.359   \n",
       "77        AG        GG           0.068           0.408           0.524   \n",
       "78        GA        AA           0.001           0.087           0.913   \n",
       "79        GA        AA           0.184           0.505           0.311   \n",
       "80        CT        TT           0.272           0.485           0.243   \n",
       "81        CT        TT           0.049           0.466           0.505   \n",
       "82        CT        TT           0.001           0.136           0.864   \n",
       "83        TG        GG           0.466           0.427           0.107   \n",
       "84        AG        GG           0.563           0.340           0.097   \n",
       "85        TC        CC           0.379           0.485           0.136   \n",
       "86        TG        GG           0.058           0.330           0.612   \n",
       "87        AG        GG           0.010           0.049           0.942   \n",
       "88        CT        TT           0.194           0.515           0.291   \n",
       "89        AG        GG           0.233           0.485           0.282   \n",
       "90        CG        GG           0.505           0.427           0.068   \n",
       "91        AG        GG           0.233           0.515           0.252   \n",
       "92        CA        AA           0.874           0.126           0.999   \n",
       "93        AG        GG           0.001           0.039           0.961   \n",
       "94        AT        TT           0.184           0.466           0.350   \n",
       "95        TC        CC           0.515           0.447           0.039   \n",
       "96        CT        TT           0.214           0.515           0.272   \n",
       "\n",
       "       Standard1     Standard2   d_value d_genotype  d_score  \n",
       "0   1.000000e+00  1.000000e-01  0.219033         CT    66.60  \n",
       "1   5.000000e-03  2.000000e+00  0.061989         CG    70.48  \n",
       "2   1.000000e-03  2.000000e+00  1.178644         GA    70.10  \n",
       "3   1.000000e-01  3.000000e+00  3.157721         CC    84.86  \n",
       "4   3.000000e+00  2.000000e+00  2.959190         GC    71.06  \n",
       "5   3.000000e-01  9.999900e+04  0.162443         GG    37.88  \n",
       "6   2.000000e+00  2.000000e-01  1.373451         AC    67.00  \n",
       "7   1.000000e-03  5.000000e+00  0.370641         CG    69.70  \n",
       "8   1.000000e-05  5.000000e-01  7.049019         CC    90.68  \n",
       "9   1.000000e-05  5.000000e-01  3.126596         GG    94.18  \n",
       "10  2.500000e+00  6.000000e-01  0.134487         CC    82.14  \n",
       "11  1.500000e-01  1.000000e+00  0.106707         GG    40.60  \n",
       "12  4.800000e+00  5.000000e+00  5.030878         CC    87.38  \n",
       "13  1.000000e-03  1.000000e-04  2.149230         CC    42.12  \n",
       "14  7.000000e-01  4.500000e-01  0.732262         AA    38.64  \n",
       "15  9.999900e+04  3.000000e-01  0.137206         GG    99.42  \n",
       "16  5.200000e-01  4.000000e-01  0.364818         GG    86.22  \n",
       "17  3.000000e-01  4.000000e-01  0.159425         GG    37.08  \n",
       "18  9.999900e+04  1.400000e-01  0.138628         TT    93.00  \n",
       "19  1.000000e+00  1.000000e-01  0.272667         GC    71.06  \n",
       "20  5.000000e-01  1.000000e+00  0.096738         TT    23.00  \n",
       "21  1.000000e-01  6.000000e-03  0.065999         TG    65.64  \n",
       "22  5.000000e-01  6.000000e-01  0.027877         TT    46.40  \n",
       "23  1.000000e-05  8.000000e-01  0.761502         GA    64.28  \n",
       "24  1.500000e-01  9.999900e+04  0.118210         GG    50.28  \n",
       "25  5.000000e-01  1.000000e+00  0.466374         AA    30.08  \n",
       "26  1.000000e-05  8.000000e-01  0.761502         GA    64.28  \n",
       "27  1.000000e-01  2.000000e-01  1.584336         AA    93.00  \n",
       "28  5.000000e-01  2.000000e+00  1.356859         AG    70.00  \n",
       "29  1.000000e-05  2.000000e+00  2.583307         GG    96.50  \n",
       "..           ...           ...       ...        ...      ...  \n",
       "67  9.999900e+04  4.000000e-01  0.097417         GG    97.86  \n",
       "68  9.999900e+04  2.500000e-01  0.009986         TT    87.38  \n",
       "69  1.000000e+00  2.000000e-01  0.432137         CT    67.96  \n",
       "70  1.000000e-05  1.000000e+00  1.560999         AA    99.02  \n",
       "71  1.000000e-05  8.000000e-01  0.950663         CC    95.92  \n",
       "72  1.000000e+00  4.000000e-01  0.448746         CT    69.32  \n",
       "73  5.000000e-02  4.500000e-02  0.760001         TT    24.28  \n",
       "74  9.999900e+04  3.000000e-01  0.087483         GG    86.40  \n",
       "75  5.000000e-01  7.000000e-01  1.998541         AA    99.98  \n",
       "76  1.000000e+00  1.000000e-01  0.383795         AG    70.30  \n",
       "77  1.000000e-04  6.000000e-01  0.575595         AG    68.16  \n",
       "78  1.000000e-01  2.000000e-01  2.409493         AA    98.26  \n",
       "79  3.500000e+00  1.000000e+00  3.503678         GG    27.36  \n",
       "80  6.450000e-01  5.000000e-01  0.647525         CC    30.88  \n",
       "81  1.500000e-01  1.000000e+00  0.213565         CT    69.32  \n",
       "82  1.300000e+00  7.000000e-01  0.073876         TT    97.28  \n",
       "83  9.000000e-01  1.000000e-06  0.220932         TG    68.54  \n",
       "84  1.000000e+00  2.000000e-01  2.034827         AA    42.52  \n",
       "85  6.000000e-02  1.000000e-05  0.057823         TC    69.70  \n",
       "86  2.000000e-01  1.200000e+00  1.354940         GG    92.24  \n",
       "87  9.999900e+04  2.000000e+04  0.088828         GG    98.84  \n",
       "88  9.999999e+06  4.000000e-01  0.379395         TT    85.82  \n",
       "89  3.000000e-01  9.999990e+05  0.323409         AG    69.70  \n",
       "90  3.000000e+00  9.999999e+06  1.674857         CC    40.20  \n",
       "91  9.999999e+06  4.000000e-02  0.043170         AG    70.30  \n",
       "92  1.000000e-01  9.999900e+04  0.034762         CC    54.96  \n",
       "93  9.999999e+06  7.000000e+00  3.099021         GG    99.22  \n",
       "94  1.000000e-07  3.000000e+00  0.456020         AT    69.32  \n",
       "95  5.000000e+00  9.999900e+04  0.162091         TT    40.60  \n",
       "96  1.000000e-06  5.000000e-01  0.114713         CT    70.30  \n",
       "\n",
       "[97 rows x 18 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ag_genotype"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### A genotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 363,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AG190604_A11_B12 finish calculate the \"a_value\".\n",
      "\n",
      "AG190604_A11_B12 finish calculate \"a_genotype\".\n",
      "\n",
      "AG190604_A11_B12 finish calculate \"a_score\". \n",
      "\n",
      "AG190604_A11_B12 finish calculate \"a_result\". \n",
      "\n",
      "AG190604_A11_B12 finish output ../result/190604/AG190604-12/AG190604_A11_B12_a_result.csv\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# get value (delta vic / delta fam)\n",
    "\n",
    "a_value = value_vic_fam.iloc[[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,\n",
    "                    24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35,\n",
    "                    48, 49, \n",
    "                    35,\n",
    "                    50, 51, 52, 53, 54, 55, 56,\n",
    "                    51, 50,\n",
    "                    57, 58, 59,\n",
    "                    72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83,\n",
    "                    96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107,\n",
    "                    120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131,\n",
    "                    144, 145, 146, 147, 148, 149, 150, 151, 153, 152, 154, 155,\n",
    "                    168, 169, 170, 171, 172, 173, 174, 175, 176, 177], [6]]\n",
    "\n",
    "a_ratio = []\n",
    "\n",
    "for row in range(len(a_value)):\n",
    "    \n",
    "    a_ratio.append(a_value.iloc[row,0])\n",
    "\n",
    "print(AG_num + ' finish calculate the \"a_value\".' + '\\n')\n",
    "\n",
    "#a_ratio\n",
    "\n",
    "\n",
    "# Get genotype\n",
    "\n",
    "a_genotype = []\n",
    "\n",
    "for row in range(len(a_value)):\n",
    "    \n",
    "    if stand1[row] > stand2[row]:\n",
    "        \n",
    "        if float(a_value.iloc[row,]) > stand1[row]:\n",
    "            \n",
    "            a_genotype.append(gene1[row])\n",
    "            \n",
    "        elif float(a_value.iloc[row,]) < stand2[row]:\n",
    "            \n",
    "            a_genotype.append(gene3[row])\n",
    "            \n",
    "        else:\n",
    "            \n",
    "            a_genotype.append(gene2[row])  \n",
    "            \n",
    "    elif stand1[row] < stand2[row]:\n",
    "        \n",
    "        if float(a_value.iloc[row,]) > stand2[row]:\n",
    "            \n",
    "            a_genotype.append(gene3[row])\n",
    "            \n",
    "        elif float(a_value.iloc[row,]) < stand1[row]:\n",
    "            \n",
    "            a_genotype.append(gene1[row])\n",
    "            \n",
    "        else:\n",
    "            \n",
    "            a_genotype.append(gene2[row])  \n",
    "    \n",
    "    \n",
    "print(AG_num + ' finish calculate \"a_genotype\".' +'\\n')\n",
    "\n",
    "#a_genotype\n",
    "\n",
    "\n",
    "\n",
    "# A score\n",
    "\n",
    "a_score = []\n",
    "\n",
    "for row in range(len(a_genotype)):\n",
    "    \n",
    "    if a_genotype[row] == gene1[row]:\n",
    "        \n",
    "        a_score.append(40*risk1[row] + 20)\n",
    "        \n",
    "    elif a_genotype[row] == gene2[row]:\n",
    "        \n",
    "        a_score.append(20*risk2[row] + 60)\n",
    "        \n",
    "    elif a_genotype[row] == gene3[row]:\n",
    "        \n",
    "        a_score.append(20*risk3[row] + 80)\n",
    "        \n",
    "print(AG_num + ' finish calculate \"a_score\". ' +'\\n')   \n",
    "\n",
    "\n",
    "# A result\n",
    "\n",
    "#############\n",
    "# (1) 細胞衰老\n",
    "\n",
    "# 1.細胞循環\n",
    "\n",
    "g1_1 = mean(a_score[0:5])\n",
    "\n",
    "# 2.DNA損傷檢查點\n",
    "\n",
    "g1_2 = mean(a_score[5:10])\n",
    "\n",
    "# 3.端粒酶活性\n",
    "\n",
    "g1_3 = mean(a_score[10:12])\n",
    "\n",
    "# 4.DNA修復\n",
    "\n",
    "g1_4 = mean(a_score[12:20])\n",
    "\n",
    "# 5.粒線體活性\n",
    "\n",
    "g1_5 = mean(a_score[20:26])\n",
    "\n",
    "# 6.對抗自由基\n",
    "\n",
    "g1_6 = mean(a_score[26:32])\n",
    "\n",
    "\n",
    "#############\n",
    "# (2) 毒物代謝\n",
    "\n",
    "# 1.肝臟\n",
    "\n",
    "g2_1 = mean(a_score[32:37])\n",
    "\n",
    "# 2.肺臟\n",
    "\n",
    "g2_2 = mean(a_score[37:40])\n",
    "\n",
    "# 3.腸道\n",
    "\n",
    "g2_3 = mean(a_score[40:43])\n",
    "\n",
    "# 4.腎臟\n",
    "\n",
    "g2_4 = mean(a_score[43:47])\n",
    "\n",
    "\n",
    "#############\n",
    "# (3) 營養代謝\n",
    "\n",
    "# 1.血管健康\n",
    "\n",
    "g3_1 = mean(a_score[47:51])\n",
    "\n",
    "# 2.蛋白質代謝\n",
    "\n",
    "g3_2 = mean(a_score[51:57])\n",
    "\n",
    "# 3.醣類代謝\n",
    "\n",
    "g3_3 = mean(a_score[57:65])\n",
    "\n",
    "# 4.脂質代謝\n",
    "\n",
    "g3_4 = mean(a_score[65:72])\n",
    "\n",
    "\n",
    "#############\n",
    "# (4) 免疫調節\n",
    "\n",
    "# 1.抗原呈現\n",
    "\n",
    "g4_1 = mean(a_score[72:77])\n",
    "\n",
    "# 2.免疫訓練\n",
    "\n",
    "g4_2 = mean(a_score[77:88])\n",
    "\n",
    "# 3.發炎因子\n",
    "\n",
    "g4_3 = mean(a_score[88:97])\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# Score of 4 category\n",
    "\n",
    "# (1) 細胞衰老\n",
    "\n",
    "g1 = mean(a_score[0:32])\n",
    "\n",
    "# (2) 毒物代謝\n",
    "\n",
    "g2 = mean(a_score[32:47])\n",
    "\n",
    "# (3) 營養代謝\n",
    "\n",
    "g3 = mean(a_score[47:72])\n",
    "\n",
    "# (4) 免疫調節\n",
    "\n",
    "g4 = mean(a_score[72:97])\n",
    "\n",
    "\n",
    "#############################################\n",
    "# Pick the bad subcategory from each category\n",
    "\n",
    "\n",
    "sub = []\n",
    "\n",
    "# (1) 細胞衰老\n",
    "\n",
    "if min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_1:\n",
    "    \n",
    "    sub.append(\"細胞循環\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_2:\n",
    "\n",
    "    sub.append(\"DNA損傷檢查點\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_3:\n",
    "\n",
    "    sub.append(\"端粒酶活性\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_4:\n",
    "\n",
    "    sub.append(\"DNA修復\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_5:\n",
    "\n",
    "    sub.append(\"粒線體活性\")\n",
    "\n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_6:\n",
    "\n",
    "    sub.append(\"對抗自由基\")\n",
    "    \n",
    "\n",
    "# (2) 毒物代謝\n",
    "\n",
    "if min(g2_1, g2_2, g2_3, g2_4) == g2_1:\n",
    "    \n",
    "    sub.append(\"肝臟\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_2:\n",
    "    \n",
    "    sub.append(\"肺臟\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_3:\n",
    "    \n",
    "    sub.append(\"腸道\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_4:\n",
    "    \n",
    "    sub.append(\"腎臟\")\n",
    "    \n",
    "\n",
    "# (3) 營養代謝\n",
    "\n",
    "if min(g3_1, g3_2, g3_3, g3_4) == g3_1:\n",
    "    \n",
    "    sub.append(\"血管健康\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_2:\n",
    "    \n",
    "    sub.append(\"蛋白質代謝\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_3:\n",
    "    \n",
    "    sub.append(\"醣類代謝\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_4:\n",
    "    \n",
    "    sub.append(\"脂質代謝\")\n",
    "    \n",
    "    \n",
    "# (4) 免疫調節\n",
    "\n",
    "if min(g4_1, g4_2, g4_3) == g4_1:\n",
    "    \n",
    "    sub.append(\"抗原呈現\")\n",
    "    \n",
    "elif min(g4_1, g4_2, g4_3) == g4_2:\n",
    "    \n",
    "    sub.append(\"免疫訓練\")\n",
    "    \n",
    "elif min(g4_1, g4_2, g4_3) == g4_3:\n",
    "    \n",
    "    sub.append(\"發炎因子\")\n",
    "\n",
    "\n",
    "print(AG_num + ' finish calculate \"a_result\". ' +'\\n')\n",
    "\n",
    "sub\n",
    "\n",
    "ag_genotype['a_value'] = a_ratio\n",
    "ag_genotype['a_genotype'] = a_genotype\n",
    "ag_genotype['a_score'] = a_score\n",
    "\n",
    "ag_genotype\n",
    "\n",
    "\n",
    "outname = AG_num + '_a_result.csv'\n",
    "\n",
    "outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)\n",
    "\n",
    "fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "ag_genotype.to_csv(fullname)\n",
    "\n",
    "print(AG_num + ' finish output ' + fullname + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AG190604_A11_B12 finish calculate the \"a_value\".\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# get value (delta vic / delta fam)\n",
    "\n",
    "a_value = value_vic_fam.iloc[[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,\n",
    "                    24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35,\n",
    "                    48, 49, \n",
    "                    35,\n",
    "                    50, 51, 52, 53, 54, 55, 56,\n",
    "                    51, 50,\n",
    "                    57, 58, 59,\n",
    "                    72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83,\n",
    "                    96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107,\n",
    "                    120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131,\n",
    "                    144, 145, 146, 147, 148, 149, 150, 151, 153, 152, 154, 155,\n",
    "                    168, 169, 170, 171, 172, 173, 174, 175, 176, 177], [6]]\n",
    "\n",
    "a_ratio = []\n",
    "\n",
    "for row in range(len(a_value)):\n",
    "    \n",
    "    a_ratio.append(a_value.iloc[row,0])\n",
    "\n",
    "print(AG_num + ' finish calculate the \"a_value\".' + '\\n')\n",
    "\n",
    "#a_ratio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 365,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AG190604_A11_B12 finish calculate \"a_genotype\".\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Get genotype\n",
    "\n",
    "a_genotype = []\n",
    "\n",
    "for row in range(len(a_value)):\n",
    "    \n",
    "    if stand1[row] > stand2[row]:\n",
    "        \n",
    "        if float(a_value.iloc[row,]) > stand1[row]:\n",
    "            \n",
    "            a_genotype.append(gene1[row])\n",
    "            \n",
    "        elif float(a_value.iloc[row,]) < stand2[row]:\n",
    "            \n",
    "            a_genotype.append(gene3[row])\n",
    "            \n",
    "        else:\n",
    "            \n",
    "            a_genotype.append(gene2[row])  \n",
    "            \n",
    "    elif stand1[row] < stand2[row]:\n",
    "        \n",
    "        if float(a_value.iloc[row,]) > stand2[row]:\n",
    "            \n",
    "            a_genotype.append(gene3[row])\n",
    "            \n",
    "        elif float(a_value.iloc[row,]) < stand1[row]:\n",
    "            \n",
    "            a_genotype.append(gene1[row])\n",
    "            \n",
    "        else:\n",
    "            \n",
    "            a_genotype.append(gene2[row])  \n",
    "    \n",
    "    \n",
    "print(AG_num + ' finish calculate \"a_genotype\".' +'\\n')\n",
    "\n",
    "#a_genotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 366,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AG190604_A11_B12 finish calculate \"a_score\". \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# A score\n",
    "\n",
    "a_score = []\n",
    "\n",
    "for row in range(len(a_genotype)):\n",
    "    \n",
    "    if a_genotype[row] == gene1[row]:\n",
    "        \n",
    "        a_score.append(40*risk1[row] + 20)\n",
    "        \n",
    "    elif a_genotype[row] == gene2[row]:\n",
    "        \n",
    "        a_score.append(20*risk2[row] + 60)\n",
    "        \n",
    "    elif a_genotype[row] == gene3[row]:\n",
    "        \n",
    "        a_score.append(20*risk3[row] + 80)\n",
    "        \n",
    "print(AG_num + ' finish calculate \"a_score\". ' +'\\n')        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 367,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AG190604_A11_B12 finish calculate \"a_result\". \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['端粒酶活性', '腎臟', '血管健康', '抗原呈現']"
      ]
     },
     "execution_count": 367,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# A result\n",
    "\n",
    "#############\n",
    "# (1) 細胞衰老\n",
    "\n",
    "# 1.細胞循環\n",
    "\n",
    "g1_1 = mean(a_score[0:5])\n",
    "\n",
    "# 2.DNA損傷檢查點\n",
    "\n",
    "g1_2 = mean(a_score[5:10])\n",
    "\n",
    "# 3.端粒酶活性\n",
    "\n",
    "g1_3 = mean(a_score[10:12])\n",
    "\n",
    "# 4.DNA修復\n",
    "\n",
    "g1_4 = mean(a_score[12:20])\n",
    "\n",
    "# 5.粒線體活性\n",
    "\n",
    "g1_5 = mean(a_score[20:26])\n",
    "\n",
    "# 6.對抗自由基\n",
    "\n",
    "g1_6 = mean(a_score[26:32])\n",
    "\n",
    "\n",
    "#############\n",
    "# (2) 毒物代謝\n",
    "\n",
    "# 1.肝臟\n",
    "\n",
    "g2_1 = mean(a_score[32:37])\n",
    "\n",
    "# 2.肺臟\n",
    "\n",
    "g2_2 = mean(a_score[37:40])\n",
    "\n",
    "# 3.腸道\n",
    "\n",
    "g2_3 = mean(a_score[40:43])\n",
    "\n",
    "# 4.腎臟\n",
    "\n",
    "g2_4 = mean(a_score[43:47])\n",
    "\n",
    "\n",
    "#############\n",
    "# (3) 營養代謝\n",
    "\n",
    "# 1.血管健康\n",
    "\n",
    "g3_1 = mean(a_score[47:51])\n",
    "\n",
    "# 2.蛋白質代謝\n",
    "\n",
    "g3_2 = mean(a_score[51:57])\n",
    "\n",
    "# 3.醣類代謝\n",
    "\n",
    "g3_3 = mean(a_score[57:65])\n",
    "\n",
    "# 4.脂質代謝\n",
    "\n",
    "g3_4 = mean(a_score[65:72])\n",
    "\n",
    "\n",
    "#############\n",
    "# (4) 免疫調節\n",
    "\n",
    "# 1.抗原呈現\n",
    "\n",
    "g4_1 = mean(a_score[72:77])\n",
    "\n",
    "# 2.免疫訓練\n",
    "\n",
    "g4_2 = mean(a_score[77:88])\n",
    "\n",
    "# 3.發炎因子\n",
    "\n",
    "g4_3 = mean(a_score[88:97])\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# Score of 4 category\n",
    "\n",
    "# (1) 細胞衰老\n",
    "\n",
    "g1 = mean(a_score[0:32])\n",
    "\n",
    "# (2) 毒物代謝\n",
    "\n",
    "g2 = mean(a_score[32:47])\n",
    "\n",
    "# (3) 營養代謝\n",
    "\n",
    "g3 = mean(a_score[47:72])\n",
    "\n",
    "# (4) 免疫調節\n",
    "\n",
    "g4 = mean(a_score[72:97])\n",
    "\n",
    "\n",
    "#############################################\n",
    "# Pick the bad subcategory from each category\n",
    "\n",
    "\n",
    "sub = []\n",
    "\n",
    "# (1) 細胞衰老\n",
    "\n",
    "if min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_1:\n",
    "    \n",
    "    sub.append(\"細胞循環\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_2:\n",
    "\n",
    "    sub.append(\"DNA損傷檢查點\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_3:\n",
    "\n",
    "    sub.append(\"端粒酶活性\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_4:\n",
    "\n",
    "    sub.append(\"DNA修復\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_5:\n",
    "\n",
    "    sub.append(\"粒線體活性\")\n",
    "\n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_6:\n",
    "\n",
    "    sub.append(\"對抗自由基\")\n",
    "    \n",
    "\n",
    "# (2) 毒物代謝\n",
    "\n",
    "if min(g2_1, g2_2, g2_3, g2_4) == g2_1:\n",
    "    \n",
    "    sub.append(\"肝臟\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_2:\n",
    "    \n",
    "    sub.append(\"肺臟\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_3:\n",
    "    \n",
    "    sub.append(\"腸道\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_4:\n",
    "    \n",
    "    sub.append(\"腎臟\")\n",
    "    \n",
    "\n",
    "# (3) 營養代謝\n",
    "\n",
    "if min(g3_1, g3_2, g3_3, g3_4) == g3_1:\n",
    "    \n",
    "    sub.append(\"血管健康\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_2:\n",
    "    \n",
    "    sub.append(\"蛋白質代謝\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_3:\n",
    "    \n",
    "    sub.append(\"醣類代謝\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_4:\n",
    "    \n",
    "    sub.append(\"脂質代謝\")\n",
    "    \n",
    "    \n",
    "# (4) 免疫調節\n",
    "\n",
    "if min(g4_1, g4_2, g4_3) == g4_1:\n",
    "    \n",
    "    sub.append(\"抗原呈現\")\n",
    "    \n",
    "elif min(g4_1, g4_2, g4_3) == g4_2:\n",
    "    \n",
    "    sub.append(\"免疫訓練\")\n",
    "    \n",
    "elif min(g4_1, g4_2, g4_3) == g4_3:\n",
    "    \n",
    "    sub.append(\"發炎因子\")\n",
    "\n",
    "\n",
    "print(AG_num + ' finish calculate \"a_result\". ' +'\\n')\n",
    "\n",
    "sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AG190604_A11_B12 finish output ../result/190604/AG190604_A11_B12/AG190604_A11_B12_a_result.csv\n",
      "\n"
     ]
    }
   ],
   "source": [
    "ag_genotype['a_value'] = a_ratio\n",
    "ag_genotype['a_genotype'] = a_genotype\n",
    "ag_genotype['a_score'] = a_score\n",
    "\n",
    "ag_genotype\n",
    "\n",
    "\n",
    "outname = AG_num + '_a_result.csv'\n",
    "\n",
    "outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)\n",
    "\n",
    "fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "ag_genotype.to_csv(fullname)\n",
    "\n",
    "print(AG_num + ' finish output ' + fullname + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 368,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AG190604_A11_B12 finish output ../result/190604/AG190604-12/AG190604_A11_B12_a_category.csv\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>category</th>\n",
       "      <th>mean</th>\n",
       "      <th>high risk subcategory</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>細胞衰老</td>\n",
       "      <td>63</td>\n",
       "      <td>端粒酶活性</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>毒物代謝</td>\n",
       "      <td>66</td>\n",
       "      <td>腎臟</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>營養代謝</td>\n",
       "      <td>70</td>\n",
       "      <td>血管健康</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>免疫調節</td>\n",
       "      <td>69</td>\n",
       "      <td>抗原呈現</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  category mean high risk subcategory\n",
       "0     細胞衰老   63                 端粒酶活性\n",
       "1     毒物代謝   66                    腎臟\n",
       "2     營養代謝   70                  血管健康\n",
       "3     免疫調節   69                  抗原呈現"
      ]
     },
     "execution_count": 368,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "category = [\"細胞衰老\", \"毒物代謝\", \"營養代謝\", \"免疫調節\"]\n",
    "\n",
    "a_mean = [format(g1, '.0f'), format(g2, '.0f'), format(g3, '.0f'), format(g4, '.0f')]\n",
    "\n",
    "a_category = pd.DataFrame(list(zip(category, a_mean, sub)),\n",
    "              columns=['category','mean', 'high risk subcategory'])\n",
    "\n",
    "\n",
    "outname = AG_num + '_a_category.csv'\n",
    "\n",
    "outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)\n",
    "\n",
    "fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "a_category.to_csv(fullname)\n",
    "\n",
    "print(AG_num + ' finish output ' + fullname + '\\n')\n",
    "\n",
    "a_category"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>category</th>\n",
       "      <th>mean</th>\n",
       "      <th>high risk subcategory</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>細胞衰老</td>\n",
       "      <td>63</td>\n",
       "      <td>端粒酶活性</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>毒物代謝</td>\n",
       "      <td>66</td>\n",
       "      <td>腎臟</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>營養代謝</td>\n",
       "      <td>70</td>\n",
       "      <td>血管健康</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>免疫調節</td>\n",
       "      <td>69</td>\n",
       "      <td>抗原呈現</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  category mean high risk subcategory\n",
       "0     細胞衰老   63                 端粒酶活性\n",
       "1     毒物代謝   66                    腎臟\n",
       "2     營養代謝   70                  血管健康\n",
       "3     免疫調節   69                  抗原呈現"
      ]
     },
     "execution_count": 293,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a_category"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### B genotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AG190604_A11_B12 finish calculate the \"b_value\".\n",
      "\n",
      "AG190604_A11_B12 finish calculate \"b_genotype\".\n",
      "\n",
      "AG190604_A11_B12 finish calculate \"b_score\". \n",
      "\n",
      "AG190604_A11_B12 finish calculate \"b_result\". \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['端粒酶活性', '蛋白質代謝', '發炎因子']"
      ]
     },
     "execution_count": 258,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get value (delta vic / delta fam)\n",
    "\n",
    "b_value = value_vic_fam.iloc[[12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23,\n",
    "                    36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47,\n",
    "                    60, 61, \n",
    "                    47,\n",
    "                    62, 63, 64, 65, 66, 67, 68,\n",
    "                    63, 62,\n",
    "                    69, 70, 71,\n",
    "                    84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95,\n",
    "                    108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119,\n",
    "                    132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143,\n",
    "                    156, 157, 158, 159, 160, 161, 162, 163, 165, 164, 166, 167,\n",
    "                    180, 181, 182, 183, 184, 185, 186, 187, 188, 189], [6]]\n",
    "\n",
    "b_ratio = []\n",
    "\n",
    "for row in range(len(b_value)):\n",
    "    \n",
    "    b_ratio.append(b_value.iloc[row,0])\n",
    "\n",
    "print(AG_num + ' finish calculate the \"b_value\".' + '\\n')\n",
    "\n",
    "\n",
    "\n",
    "# Get genotype\n",
    "\n",
    "b_genotype = []\n",
    "\n",
    "for row in range(len(b_value)):\n",
    "    \n",
    "    if stand1[row] > stand2[row]:\n",
    "        \n",
    "        if float(b_value.iloc[row,]) > stand1[row]:\n",
    "            \n",
    "            b_genotype.append(gene1[row])\n",
    "            \n",
    "        elif float(b_value.iloc[row,]) < stand2[row]:\n",
    "            \n",
    "            b_genotype.append(gene3[row])\n",
    "            \n",
    "        else:\n",
    "            \n",
    "            b_genotype.append(gene2[row])  \n",
    "            \n",
    "    elif stand1[row] < stand2[row]:\n",
    "        \n",
    "        if float(b_value.iloc[row,]) > stand2[row]:\n",
    "            \n",
    "            b_genotype.append(gene3[row])\n",
    "            \n",
    "        elif float(b_value.iloc[row,]) < stand1[row]:\n",
    "            \n",
    "            b_genotype.append(gene1[row])\n",
    "            \n",
    "        else:\n",
    "            \n",
    "            b_genotype.append(gene2[row])  \n",
    "    \n",
    "    \n",
    "print(AG_num + ' finish calculate \"b_genotype\".' +'\\n')\n",
    "\n",
    "#b_genotype\n",
    "\n",
    "\n",
    "\n",
    "# B score\n",
    "\n",
    "b_score = []\n",
    "\n",
    "for row in range(len(b_genotype)):\n",
    "    \n",
    "    if b_genotype[row] == gene1[row]:\n",
    "        \n",
    "        b_score.append(40*risk1[row] + 20)\n",
    "        \n",
    "    elif b_genotype[row] == gene2[row]:\n",
    "        \n",
    "        b_score.append(20*risk2[row] + 60)\n",
    "        \n",
    "    elif b_genotype[row] == gene3[row]:\n",
    "        \n",
    "        b_score.append(20*risk3[row] + 80)\n",
    "        \n",
    "print(AG_num + ' finish calculate \"b_score\". ' +'\\n')   \n",
    "\n",
    "\n",
    "\n",
    "# B result\n",
    "\n",
    "#############\n",
    "# (1) 細胞衰老\n",
    "\n",
    "# 1.細胞循環\n",
    "\n",
    "g1_1 = mean(b_score[0:5])\n",
    "\n",
    "# 2.DNA損傷檢查點\n",
    "\n",
    "g1_2 = mean(b_score[5:10])\n",
    "\n",
    "# 3.端粒酶活性\n",
    "\n",
    "g1_3 = mean(b_score[10:12])\n",
    "\n",
    "# 4.DNA修復\n",
    "\n",
    "g1_4 = mean(b_score[12:20])\n",
    "\n",
    "# 5.粒線體活性\n",
    "\n",
    "g1_5 = mean(b_score[20:26])\n",
    "\n",
    "# 6.對抗自由基\n",
    "\n",
    "g1_6 = mean(b_score[26:32])\n",
    "\n",
    "\n",
    "#############\n",
    "# (2) 毒物代謝\n",
    "\n",
    "# 1.肝臟\n",
    "\n",
    "g2_1 = mean(b_score[32:37])\n",
    "\n",
    "# 2.肺臟\n",
    "\n",
    "g2_2 = mean(b_score[37:40])\n",
    "\n",
    "# 3.腸道\n",
    "\n",
    "g2_3 = mean(b_score[40:43])\n",
    "\n",
    "# 4.腎臟\n",
    "\n",
    "g2_4 = mean(b_score[43:47])\n",
    "\n",
    "\n",
    "#############\n",
    "# (3) 營養代謝\n",
    "\n",
    "# 1.血管健康\n",
    "\n",
    "g3_1 = mean(b_score[47:51])\n",
    "\n",
    "# 2.蛋白質代謝\n",
    "\n",
    "g3_2 = mean(b_score[51:57])\n",
    "\n",
    "# 3.醣類代謝\n",
    "\n",
    "g3_3 = mean(b_score[57:65])\n",
    "\n",
    "# 4.脂質代謝\n",
    "\n",
    "g3_4 = mean(b_score[65:72])\n",
    "\n",
    "\n",
    "#############\n",
    "# (4) 免疫調節\n",
    "\n",
    "# 1.抗原呈現\n",
    "\n",
    "g4_1 = mean(b_score[72:77])\n",
    "\n",
    "# 2.免疫訓練\n",
    "\n",
    "g4_2 = mean(b_score[77:88])\n",
    "\n",
    "# 3.發炎因子\n",
    "\n",
    "g4_3 = mean(b_score[88:97])\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# Score of 4 category\n",
    "\n",
    "# (1) 細胞衰老\n",
    "\n",
    "g1 = mean(b_score[0:32])\n",
    "\n",
    "# (2) 毒物代謝\n",
    "\n",
    "g2 = mean(b_score[32:47])\n",
    "\n",
    "# (3) 營養代謝\n",
    "\n",
    "g3 = mean(b_score[47:72])\n",
    "\n",
    "# (4) 免疫調節\n",
    "\n",
    "g4 = mean(b_score[72:97])\n",
    "\n",
    "\n",
    "#############################################\n",
    "# Pick the bad subcategory from each category\n",
    "\n",
    "\n",
    "sub = []\n",
    "\n",
    "# (1) 細胞衰老\n",
    "\n",
    "if min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_1:\n",
    "    \n",
    "    sub.append(\"細胞循環\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_2:\n",
    "\n",
    "    sub.append(\"DNA損傷檢查點\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_3:\n",
    "\n",
    "    sub.append(\"端粒酶活性\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_4:\n",
    "\n",
    "    sub.append(\"DNA修復\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_5:\n",
    "\n",
    "    sub.append(\"粒線體活性\")\n",
    "\n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_6:\n",
    "\n",
    "    sub.append(\"對抗自由基\")\n",
    "    \n",
    "\n",
    "# (2) 毒物代謝\n",
    "\n",
    "if min(g2_1, g2_2, g2_3, g2_4) == g2_1:\n",
    "    \n",
    "    sub.append(\"肝臟\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_2:\n",
    "    \n",
    "    sub.append(\"肺臟\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_3:\n",
    "    \n",
    "    sub.append(\"腸道\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_4:\n",
    "    \n",
    "    sub.append(\"腎臟\")\n",
    "    \n",
    "\n",
    "# (3) 營養代謝\n",
    "\n",
    "if min(g3_1, g3_2, g3_3, g3_4) == g3_1:\n",
    "    \n",
    "    sub.append(\"血管健康\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_2:\n",
    "    \n",
    "    sub.append(\"蛋白質代謝\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_3:\n",
    "    \n",
    "    sub.append(\"醣類代謝\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_4:\n",
    "    \n",
    "    sub.append(\"脂質代謝\")\n",
    "    \n",
    "    \n",
    "# (4) 免疫調節\n",
    "\n",
    "if min(g4_1, g4_2, g4_3) == g4_1:\n",
    "    \n",
    "    sub.append(\"抗原呈現\")\n",
    "    \n",
    "elif min(g4_1, g4_2, g4_3) == g4_2:\n",
    "    \n",
    "    sub.append(\"免疫訓練\")\n",
    "    \n",
    "elif min(g4_1, g4_2, g4_3) == g4_3:\n",
    "    \n",
    "    sub.append(\"發炎因子\")\n",
    "\n",
    "\n",
    "print(AG_num + ' finish calculate \"b_result\". ' +'\\n')\n",
    "\n",
    "sub\n",
    "\n",
    "####\n",
    "\n",
    "ag_genotype['b_value'] = b_ratio\n",
    "ag_genotype['b_genotype'] = b_genotype\n",
    "ag_genotype['b_score'] = b_score\n",
    "\n",
    "ag_genotype\n",
    "\n",
    "\n",
    "outname = AG_num + '_b_result.csv'\n",
    "\n",
    "outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)\n",
    "\n",
    "fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "ag_genotype.to_csv(fullname)\n",
    "\n",
    "print(AG_num + ' finish output ' + fullname + '\\n')\n",
    "\n",
    "#####\n",
    "\n",
    "category = [\"細胞衰老\", \"毒物代謝\", \"營養代謝\", \"免疫調節\"]\n",
    "\n",
    "b_mean = [format(g1, '.0f'), format(g2, '.0f'), format(g3, '.0f'), format(g4, '.0f')]\n",
    "\n",
    "b_category = pd.DataFrame(list(zip(category, b_mean, sub)),\n",
    "              columns=['category','mean', 'high risk subcategory'])\n",
    "\n",
    "\n",
    "outname = AG_num + '_b_category.csv'\n",
    "\n",
    "outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)\n",
    "\n",
    "fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "b_category.to_csv(fullname)\n",
    "\n",
    "print(AG_num + ' finish output ' + fullname + '\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### C genotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AG190604_A11_B12 finish calculate the \"c_value\".\n",
      "\n",
      "AG190604_A11_B12 finish calculate \"c_genotype\".\n",
      "\n",
      "AG190604_A11_B12 finish calculate \"c_score\". \n",
      "\n",
      "AG190604_A11_B12 finish calculate \"c_result\". \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['端粒酶活性', '肝臟', '醣類代謝', '發炎因子']"
      ]
     },
     "execution_count": 259,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get value (delta vic / delta fam)\n",
    "\n",
    "c_value = value_vic_fam.iloc[[192, 193, 194, 195, 196, 197, 198, 199, 200, 201, 202, 203,\n",
    "                    216, 217, 218, 219, 220, 221, 222, 223, 224, 225, 226, 227,\n",
    "                    240, 241, \n",
    "                    227,\n",
    "                    242, 243, 244, 245, 246, 247, 248,\n",
    "                    243, 242,\n",
    "                    249, 250, 251,\n",
    "                    264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 275,\n",
    "                    288, 289, 290, 291, 292, 293, 294, 295, 296, 297, 298, 299,\n",
    "                    312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323,\n",
    "                    336, 337, 338, 339, 340, 341, 342, 343, 345, 344, 346, 347,\n",
    "                    360, 361, 362, 363, 364, 365, 366, 367, 368, 369], [6]]\n",
    "\n",
    "c_ratio = []\n",
    "\n",
    "for row in range(len(c_value)):\n",
    "    \n",
    "    c_ratio.append(c_value.iloc[row,0])\n",
    "\n",
    "print(AG_num + ' finish calculate the \"c_value\".' + '\\n')\n",
    "\n",
    "\n",
    "# Get genotype\n",
    "\n",
    "c_genotype = []\n",
    "\n",
    "for row in range(len(c_value)):\n",
    "    \n",
    "    if stand1[row] > stand2[row]:\n",
    "        \n",
    "        if float(c_value.iloc[row,]) > stand1[row]:\n",
    "            \n",
    "            c_genotype.append(gene1[row])\n",
    "            \n",
    "        elif float(c_value.iloc[row,]) < stand2[row]:\n",
    "            \n",
    "            c_genotype.append(gene3[row])\n",
    "            \n",
    "        else:\n",
    "            \n",
    "            c_genotype.append(gene2[row])  \n",
    "            \n",
    "    elif stand1[row] < stand2[row]:\n",
    "        \n",
    "        if float(c_value.iloc[row,]) > stand2[row]:\n",
    "            \n",
    "            c_genotype.append(gene3[row])\n",
    "            \n",
    "        elif float(c_value.iloc[row,]) < stand1[row]:\n",
    "            \n",
    "            c_genotype.append(gene1[row])\n",
    "            \n",
    "        else:\n",
    "            \n",
    "            c_genotype.append(gene2[row])  \n",
    "    \n",
    "    \n",
    "print(AG_num + ' finish calculate \"c_genotype\".' +'\\n')\n",
    "\n",
    "#c_genotype\n",
    "\n",
    "\n",
    "\n",
    "# C score\n",
    "\n",
    "c_score = []\n",
    "\n",
    "for row in range(len(c_genotype)):\n",
    "    \n",
    "    if c_genotype[row] == gene1[row]:\n",
    "        \n",
    "        c_score.append(40*risk1[row] + 20)\n",
    "        \n",
    "    elif c_genotype[row] == gene2[row]:\n",
    "        \n",
    "        c_score.append(20*risk2[row] + 60)\n",
    "        \n",
    "    elif c_genotype[row] == gene3[row]:\n",
    "        \n",
    "        c_score.append(20*risk3[row] + 80)\n",
    "        \n",
    "print(AG_num + ' finish calculate \"c_score\". ' +'\\n')   \n",
    "\n",
    "\n",
    "\n",
    "# C result\n",
    "\n",
    "#############\n",
    "# (1) 細胞衰老\n",
    "\n",
    "# 1.細胞循環\n",
    "\n",
    "g1_1 = mean(c_score[0:5])\n",
    "\n",
    "# 2.DNA損傷檢查點\n",
    "\n",
    "g1_2 = mean(c_score[5:10])\n",
    "\n",
    "# 3.端粒酶活性\n",
    "\n",
    "g1_3 = mean(c_score[10:12])\n",
    "\n",
    "# 4.DNA修復\n",
    "\n",
    "g1_4 = mean(c_score[12:20])\n",
    "\n",
    "# 5.粒線體活性\n",
    "\n",
    "g1_5 = mean(c_score[20:26])\n",
    "\n",
    "# 6.對抗自由基\n",
    "\n",
    "g1_6 = mean(c_score[26:32])\n",
    "\n",
    "\n",
    "#############\n",
    "# (2) 毒物代謝\n",
    "\n",
    "# 1.肝臟\n",
    "\n",
    "g2_1 = mean(c_score[32:37])\n",
    "\n",
    "# 2.肺臟\n",
    "\n",
    "g2_2 = mean(c_score[37:40])\n",
    "\n",
    "# 3.腸道\n",
    "\n",
    "g2_3 = mean(c_score[40:43])\n",
    "\n",
    "# 4.腎臟\n",
    "\n",
    "g2_4 = mean(c_score[43:47])\n",
    "\n",
    "\n",
    "#############\n",
    "# (3) 營養代謝\n",
    "\n",
    "# 1.血管健康\n",
    "\n",
    "g3_1 = mean(c_score[47:51])\n",
    "\n",
    "# 2.蛋白質代謝\n",
    "\n",
    "g3_2 = mean(c_score[51:57])\n",
    "\n",
    "# 3.醣類代謝\n",
    "\n",
    "g3_3 = mean(c_score[57:65])\n",
    "\n",
    "# 4.脂質代謝\n",
    "\n",
    "g3_4 = mean(c_score[65:72])\n",
    "\n",
    "\n",
    "#############\n",
    "# (4) 免疫調節\n",
    "\n",
    "# 1.抗原呈現\n",
    "\n",
    "g4_1 = mean(c_score[72:77])\n",
    "\n",
    "# 2.免疫訓練\n",
    "\n",
    "g4_2 = mean(c_score[77:88])\n",
    "\n",
    "# 3.發炎因子\n",
    "\n",
    "g4_3 = mean(c_score[88:97])\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# Score of 4 category\n",
    "\n",
    "# (1) 細胞衰老\n",
    "\n",
    "g1 = mean(c_score[0:32])\n",
    "\n",
    "# (2) 毒物代謝\n",
    "\n",
    "g2 = mean(c_score[32:47])\n",
    "\n",
    "# (3) 營養代謝\n",
    "\n",
    "g3 = mean(c_score[47:72])\n",
    "\n",
    "# (4) 免疫調節\n",
    "\n",
    "g4 = mean(c_score[72:97])\n",
    "\n",
    "\n",
    "#############################################\n",
    "# Pick the bad subcategory from each category\n",
    "\n",
    "\n",
    "sub = []\n",
    "\n",
    "# (1) 細胞衰老\n",
    "\n",
    "if min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_1:\n",
    "    \n",
    "    sub.append(\"細胞循環\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_2:\n",
    "\n",
    "    sub.append(\"DNA損傷檢查點\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_3:\n",
    "\n",
    "    sub.append(\"端粒酶活性\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_4:\n",
    "\n",
    "    sub.append(\"DNA修復\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_5:\n",
    "\n",
    "    sub.append(\"粒線體活性\")\n",
    "\n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_6:\n",
    "\n",
    "    sub.append(\"對抗自由基\")\n",
    "    \n",
    "\n",
    "# (2) 毒物代謝\n",
    "\n",
    "if min(g2_1, g2_2, g2_3, g2_4) == g2_1:\n",
    "    \n",
    "    sub.append(\"肝臟\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_2:\n",
    "    \n",
    "    sub.append(\"肺臟\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_3:\n",
    "    \n",
    "    sub.append(\"腸道\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_4:\n",
    "    \n",
    "    sub.append(\"腎臟\")\n",
    "    \n",
    "\n",
    "# (3) 營養代謝\n",
    "\n",
    "if min(g3_1, g3_2, g3_3, g3_4) == g3_1:\n",
    "    \n",
    "    sub.append(\"血管健康\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_2:\n",
    "    \n",
    "    sub.append(\"蛋白質代謝\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_3:\n",
    "    \n",
    "    sub.append(\"醣類代謝\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_4:\n",
    "    \n",
    "    sub.append(\"脂質代謝\")\n",
    "    \n",
    "    \n",
    "# (4) 免疫調節\n",
    "\n",
    "if min(g4_1, g4_2, g4_3) == g4_1:\n",
    "    \n",
    "    sub.append(\"抗原呈現\")\n",
    "    \n",
    "elif min(g4_1, g4_2, g4_3) == g4_2:\n",
    "    \n",
    "    sub.append(\"免疫訓練\")\n",
    "    \n",
    "elif min(g4_1, g4_2, g4_3) == g4_3:\n",
    "    \n",
    "    sub.append(\"發炎因子\")\n",
    "\n",
    "\n",
    "print(AG_num + ' finish calculate \"c_result\". ' +'\\n')\n",
    "\n",
    "sub\n",
    "\n",
    "\n",
    "ag_genotype['c_value'] = c_ratio\n",
    "ag_genotype['c_genotype'] = c_genotype\n",
    "ag_genotype['c_score'] = c_score\n",
    "\n",
    "ag_genotype\n",
    "\n",
    "\n",
    "outname = AG_num + '_c_result.csv'\n",
    "\n",
    "outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)\n",
    "\n",
    "fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "ag_genotype.to_csv(fullname)\n",
    "\n",
    "print(AG_num + ' finish output ' + fullname + '\\n')\n",
    "\n",
    "\n",
    "#######################\n",
    "# output c_category.csv\n",
    "    \n",
    "category = [\"細胞衰老\", \"毒物代謝\", \"營養代謝\", \"免疫調節\"]\n",
    "\n",
    "c_mean = [format(g1, '.0f'), format(g2, '.0f'), format(g3, '.0f'), format(g4, '.0f')]\n",
    "\n",
    "c_category = pd.DataFrame(list(zip(category, c_mean, sub)),\n",
    "                columns=['category','mean', 'high risk subcategory'])\n",
    "\n",
    "\n",
    "outname = AG_num + '_c_category.csv'\n",
    "\n",
    "outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)\n",
    "\n",
    "fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "c_category.to_csv(fullname)\n",
    "\n",
    "print(AG_num + ' finish output ' + fullname + '\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### D genotype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AG190604_A11_B12 finish output ../result/AG190604_A11_B12/AG190604_A11_B12_d_value.csv\n",
      "\n",
      "AG190604_A11_B12 finish calculate \"d_genotype\".\n",
      "\n",
      "AG190604_A11_B12 finish calculate \"d_score\". \n",
      "\n",
      "AG190604_A11_B12 finish calculate \"d_result\". \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['端粒酶活性', '肝臟', '醣類代謝', '發炎因子']"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get value (delta vic / delta fam)\n",
    "\n",
    "d_value = value_vic_fam.iloc[[204, 205, 206, 207, 208, 209, 210, 211, 212, 213, 214, 215,\n",
    "                    228, 229, 230, 231, 232, 233, 234, 235, 236, 237, 238, 239,\n",
    "                    252, 253, \n",
    "                    239,\n",
    "                    254, 255, 256, 257, 258, 259, 260,\n",
    "                    255, 254,\n",
    "                    261, 262, 263,\n",
    "                    276, 277, 278, 279, 280, 281, 282, 283, 284, 285, 286, 287,\n",
    "                    300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 310, 311,\n",
    "                    324, 325, 326, 327, 328, 329, 330, 331, 332, 333, 334, 335,\n",
    "                    348, 349, 350, 351, 352, 353, 354, 355, 357, 356, 358, 359,\n",
    "                    372, 373, 374, 375, 376, 377, 378, 379, 380, 381], [6]]\n",
    "\n",
    "d_ratio = []\n",
    "\n",
    "for row in range(len(d_value)):\n",
    "    \n",
    "    d_ratio.append(d_value.iloc[row,0])\n",
    "\n",
    "print(AG_num + ' finish calculate the \"d_value\".' + '\\n')\n",
    "\n",
    "\n",
    "# Get genotype\n",
    "\n",
    "d_genotype = []\n",
    "\n",
    "for row in range(len(d_value)):\n",
    "    \n",
    "    if stand1[row] > stand2[row]:\n",
    "        \n",
    "        if float(d_value.iloc[row,]) > stand1[row]:\n",
    "            \n",
    "            d_genotype.append(gene1[row])\n",
    "            \n",
    "        elif float(d_value.iloc[row,]) < stand2[row]:\n",
    "            \n",
    "            d_genotype.append(gene3[row])\n",
    "            \n",
    "        else:\n",
    "            \n",
    "            d_genotype.append(gene2[row])  \n",
    "            \n",
    "    elif stand1[row] < stand2[row]:\n",
    "        \n",
    "        if float(d_value.iloc[row,]) > stand2[row]:\n",
    "            \n",
    "            d_genotype.append(gene3[row])\n",
    "            \n",
    "        elif float(d_value.iloc[row,]) < stand1[row]:\n",
    "            \n",
    "            d_genotype.append(gene1[row])\n",
    "            \n",
    "        else:\n",
    "            \n",
    "            d_genotype.append(gene2[row])  \n",
    "    \n",
    "    \n",
    "print(AG_num + ' finish calculate \"d_genotype\".' +'\\n')\n",
    "\n",
    "#d_genotype\n",
    "\n",
    "\n",
    "\n",
    "# D score\n",
    "\n",
    "d_score = []\n",
    "\n",
    "for row in range(len(d_genotype)):\n",
    "    \n",
    "    if d_genotype[row] == gene1[row]:\n",
    "        \n",
    "        d_score.append(40*risk1[row] + 20)\n",
    "        \n",
    "    elif d_genotype[row] == gene2[row]:\n",
    "        \n",
    "        d_score.append(20*risk2[row] + 60)\n",
    "        \n",
    "    elif d_genotype[row] == gene3[row]:\n",
    "        \n",
    "        d_score.append(20*risk3[row] + 80)\n",
    "        \n",
    "print(AG_num + ' finish calculate \"d_score\". ' +'\\n')   \n",
    "\n",
    "\n",
    "\n",
    "# D result\n",
    "\n",
    "#############\n",
    "# (1) 細胞衰老\n",
    "\n",
    "# 1.細胞循環\n",
    "\n",
    "g1_1 = mean(d_score[0:5])\n",
    "\n",
    "# 2.DNA損傷檢查點\n",
    "\n",
    "g1_2 = mean(d_score[5:10])\n",
    "\n",
    "# 3.端粒酶活性\n",
    "\n",
    "g1_3 = mean(d_score[10:12])\n",
    "\n",
    "# 4.DNA修復\n",
    "\n",
    "g1_4 = mean(d_score[12:20])\n",
    "\n",
    "# 5.粒線體活性\n",
    "\n",
    "g1_5 = mean(d_score[20:26])\n",
    "\n",
    "# 6.對抗自由基\n",
    "\n",
    "g1_6 = mean(d_score[26:32])\n",
    "\n",
    "\n",
    "#############\n",
    "# (2) 毒物代謝\n",
    "\n",
    "# 1.肝臟\n",
    "\n",
    "g2_1 = mean(d_score[32:37])\n",
    "\n",
    "# 2.肺臟\n",
    "\n",
    "g2_2 = mean(d_score[37:40])\n",
    "\n",
    "# 3.腸道\n",
    "\n",
    "g2_3 = mean(d_score[40:43])\n",
    "\n",
    "# 4.腎臟\n",
    "\n",
    "g2_4 = mean(d_score[43:47])\n",
    "\n",
    "\n",
    "#############\n",
    "# (3) 營養代謝\n",
    "\n",
    "# 1.血管健康\n",
    "\n",
    "g3_1 = mean(d_score[47:51])\n",
    "\n",
    "# 2.蛋白質代謝\n",
    "\n",
    "g3_2 = mean(d_score[51:57])\n",
    "\n",
    "# 3.醣類代謝\n",
    "\n",
    "g3_3 = mean(d_score[57:65])\n",
    "\n",
    "# 4.脂質代謝\n",
    "\n",
    "g3_4 = mean(d_score[65:72])\n",
    "\n",
    "\n",
    "#############\n",
    "# (4) 免疫調節\n",
    "\n",
    "# 1.抗原呈現\n",
    "\n",
    "g4_1 = mean(d_score[72:77])\n",
    "\n",
    "# 2.免疫訓練\n",
    "\n",
    "g4_2 = mean(d_score[77:88])\n",
    "\n",
    "# 3.發炎因子\n",
    "\n",
    "g4_3 = mean(d_score[88:97])\n",
    "\n",
    "\n",
    "\n",
    "#####################\n",
    "# Score of 4 category\n",
    "\n",
    "# (1) 細胞衰老\n",
    "\n",
    "g1 = mean(d_score[0:32])\n",
    "\n",
    "# (2) 毒物代謝\n",
    "\n",
    "g2 = mean(d_score[32:47])\n",
    "\n",
    "# (3) 營養代謝\n",
    "\n",
    "g3 = mean(d_score[47:72])\n",
    "\n",
    "# (4) 免疫調節\n",
    "\n",
    "g4 = mean(d_score[72:97])\n",
    "\n",
    "\n",
    "#############################################\n",
    "# Pick the bad subcategory from each category\n",
    "\n",
    "\n",
    "sub = []\n",
    "\n",
    "# (1) 細胞衰老\n",
    "\n",
    "if min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_1:\n",
    "    \n",
    "    sub.append(\"細胞循環\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_2:\n",
    "\n",
    "    sub.append(\"DNA損傷檢查點\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_3:\n",
    "\n",
    "    sub.append(\"端粒酶活性\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_4:\n",
    "\n",
    "    sub.append(\"DNA修復\")\n",
    "    \n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_5:\n",
    "\n",
    "    sub.append(\"粒線體活性\")\n",
    "\n",
    "elif min(g1_1, g1_2, g1_3, g1_4, g1_5, g1_6) == g1_6:\n",
    "\n",
    "    sub.append(\"對抗自由基\")\n",
    "    \n",
    "\n",
    "# (2) 毒物代謝\n",
    "\n",
    "if min(g2_1, g2_2, g2_3, g2_4) == g2_1:\n",
    "    \n",
    "    sub.append(\"肝臟\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_2:\n",
    "    \n",
    "    sub.append(\"肺臟\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_3:\n",
    "    \n",
    "    sub.append(\"腸道\")\n",
    "    \n",
    "elif min(g2_1, g2_2, g2_3, g2_4) == g2_4:\n",
    "    \n",
    "    sub.append(\"腎臟\")\n",
    "    \n",
    "\n",
    "# (3) 營養代謝\n",
    "\n",
    "if min(g3_1, g3_2, g3_3, g3_4) == g3_1:\n",
    "    \n",
    "    sub.append(\"血管健康\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_2:\n",
    "    \n",
    "    sub.append(\"蛋白質代謝\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_3:\n",
    "    \n",
    "    sub.append(\"醣類代謝\")\n",
    "    \n",
    "elif min(g3_1, g3_2, g3_3, g3_4) == g3_4:\n",
    "    \n",
    "    sub.append(\"脂質代謝\")\n",
    "    \n",
    "    \n",
    "# (4) 免疫調節\n",
    "\n",
    "if min(g4_1, g4_2, g4_3) == g4_1:\n",
    "    \n",
    "    sub.append(\"抗原呈現\")\n",
    "    \n",
    "elif min(g4_1, g4_2, g4_3) == g4_2:\n",
    "    \n",
    "    sub.append(\"免疫訓練\")\n",
    "    \n",
    "elif min(g4_1, g4_2, g4_3) == g4_3:\n",
    "    \n",
    "    sub.append(\"發炎因子\")\n",
    "\n",
    "\n",
    "print(AG_num + ' finish calculate \"d_result\". ' +'\\n')\n",
    "\n",
    "sub\n",
    "\n",
    "\n",
    "ag_genotype['d_value'] = d_ratio\n",
    "ag_genotype['d_genotype'] = d_genotype\n",
    "ag_genotype['d_score'] = d_score\n",
    "\n",
    "ag_genotype\n",
    "\n",
    "\n",
    "outname = AG_num + '_d_result.csv'\n",
    "\n",
    "outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)\n",
    "\n",
    "fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "ag_genotype.to_csv(fullname)\n",
    "\n",
    "print(AG_num + ' finish output ' + fullname + '\\n')\n",
    "\n",
    "\n",
    "#######################\n",
    "# output d_category.csv\n",
    "    \n",
    "category = [\"細胞衰老\", \"毒物代謝\", \"營養代謝\", \"免疫調節\"]\n",
    "\n",
    "d_mean = [format(g1, '.0f'), format(g2, '.0f'), format(g3, '.0f'), format(g4, '.0f')]\n",
    "\n",
    "d_category = pd.DataFrame(list(zip(category, b_mean, sub)),\n",
    "                columns=['category','mean', 'high risk subcategory'])\n",
    "\n",
    "\n",
    "outname = AG_num + '_d_category.csv'\n",
    "\n",
    "outdir = '../result/' + AG_sample[2:8] + '/' + AG_sample + '/'\n",
    "if not os.path.exists(outdir):\n",
    "    os.makedirs(outdir)\n",
    "\n",
    "fullname = os.path.join(outdir, outname)    \n",
    "\n",
    "d_category.to_csv(fullname)\n",
    "\n",
    "print(AG_num + ' finish output ' + fullname + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
